<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2019-12-16" />

<title>Untitled</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Untitled</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">December 16, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-12-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rna-seq-dhx38/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38treed602320dfafe088c010d645b5a782854b0cc40e4targetblankd602320a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/rna-seq-dhx38/tree/d602320dfafe088c010d645b5a782854b0cc40e4" target="_blank">d602320</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38treed602320dfafe088c010d645b5a782854b0cc40e4targetblankd602320a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/20191216_DifferentialExpression_Splicing.Rmd
    Untracked:  data/GO_splicing_ids.fromAmiGO_online_search.tab
    Untracked:  dsdist.

Unstaged changes:
    Modified:   analysis/20191203_DiffExpression.Rmd
    Modified:   analysis/20191203_DifferentialSplicing.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/dbbdff32af42a7bdbc1cd9c6d91adbb7bb55845f/analysis/20191203_DiffExpression.Rmd" target="_blank">dbbdff3</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/rna-seq-dhx38/dbbdff32af42a7bdbc1cd9c6d91adbb7bb55845f/docs/20191203_DiffExpression.html" target="_blank">dbbdff3</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/a0c762759c7f066dc175ebf7988a53401ad16cc8/analysis/20191203_DiffExpression.Rmd" target="_blank">a0c7627</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/rna-seq-dhx38/a0c762759c7f066dc175ebf7988a53401ad16cc8/docs/20191203_DiffExpression.html" target="_blank">a0c7627</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/analysis/20191203_DiffExpression.Rmd" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
<td>
initialize site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/rna-seq-dhx38/c1d660db6d38073970096cd71db8fa190693c0d4/docs/20191203_DiffExpression.html" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
<td>
initialize site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Exploratory differential expression analysis comparing RNA-seq from iPSC derived neurons from healthy control vs patient with biallelic mutations in DHX38 that may affect splicing and gene expression. In particular, worth checking whether DHX38 is differentially expressed.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Raw RNA-seq data has been aligned and collapsed into gene count table (rows of genes, columns of samples, each cell is exonic read counts) using code in <code>code</code> section of this repo, creating a count matrix in <code>output</code>. Here I will process that data with edgeR approach that I have used previously.</p>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(edgeR)
library(corrplot)
library(gplots)
library(biomaRt)
library(ggrepel)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)</code></pre>
<p>And read in the count table</p>
<pre class="r"><code>CountTable &lt;- read.table(&#39;../output/GeneExpressionCountTable.subread.txt.gz&#39;, header=T, check.names=FALSE, skip=1)


CountTable.matrix &lt;- CountTable %&gt;%
  dplyr::select(-c(&quot;Chr&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;Strand&quot;, &quot;Length&quot;)) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;% as.matrix()

#Preview table
kable(CountTable.matrix[1:10,])</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Affected_3</th>
<th align="right">Healthy_2</th>
<th align="right">Affected_1</th>
<th align="right">Affected_2</th>
<th align="right">Healthy_1</th>
<th align="right">Healthy_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000223972</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>ENSG00000227232</td>
<td align="right">36</td>
<td align="right">28</td>
<td align="right">59</td>
<td align="right">57</td>
<td align="right">32</td>
<td align="right">38</td>
</tr>
<tr class="odd">
<td>ENSG00000278267</td>
<td align="right">11</td>
<td align="right">9</td>
<td align="right">13</td>
<td align="right">12</td>
<td align="right">14</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td>ENSG00000243485</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000284332</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000237613</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000268020</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000240361</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000186092</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000238009</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Plot total exonic gene counts per sample</p>
<pre class="r"><code>CountTable.matrix %&gt;% colSums() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(&quot;Sample&quot;) %&gt;%
  ggplot(aes(x=Sample, y=./1000000)) +
  geom_col() +
  ylab(&quot;Exonic read count (Millions)&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-3-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Check correlations between samples</p>
<pre class="r"><code># CountTable to CountsPerMillion
cpm &lt;- cpm(CountTable.matrix, log=TRUE, prior.count=0.5)
kable(cpm[1:10,])</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">A</th>
<th align="left">ffected_3</th>
<th>Healthy_2 A</th>
<th>ffected_1 A</th>
<th align="left">ffected_2</th>
<th align="left">Healthy_1</th>
<th align="left">Healthy_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">ENSG00000223972</td>
<td align="left">-5.660565</td>
<td>-2.6130864</td>
<td>-3.528788</td>
<td align="left">-4.317617</td>
<td align="left">-5.6605654</td>
<td align="left">-3.9519176</td>
</tr>
<tr class="even">
<td align="right">ENSG00000227232</td>
<td align="left">0.384709</td>
<td>0.4444315</td>
<td>0.994584</td>
<td align="left">0.808530</td>
<td align="left">0.8661057</td>
<td align="left">0.7857677</td>
</tr>
<tr class="odd">
<td align="right">ENSG00000278267</td>
<td align="left">-1.276970</td>
<td>-1.1494150</td>
<td>-1.137837</td>
<td align="left">-1.379587</td>
<td align="left">-0.3065598</td>
<td align="left">-0.2739689</td>
</tr>
<tr class="even">
<td align="right">ENSG00000243485</td>
<td align="left">-5.660565</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-3.634338</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
<tr class="odd">
<td align="right">ENSG00000284332</td>
<td align="left">-5.660565</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-5.660565</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
<tr class="even">
<td align="right">ENSG00000237613</td>
<td align="left">-5.660565</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-5.660565</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
<tr class="odd">
<td align="right">ENSG00000268020</td>
<td align="left">-5.660565</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-5.660565</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
<tr class="even">
<td align="right">ENSG00000240361</td>
<td align="left">-5.660565</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-5.660565</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
<tr class="odd">
<td align="right">ENSG00000186092</td>
<td align="left">-5.660565</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-5.660565</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
<tr class="even">
<td align="right">ENSG00000238009</td>
<td align="left">-3.454751</td>
<td>-5.6605654</td>
<td>-5.660565</td>
<td align="left">-5.660565</td>
<td align="left">-5.6605654</td>
<td align="left">-5.6605654</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#plot cpm
median_log2_cpm &lt;- apply(cpm, 1, median)
hist(median_log2_cpm)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-4-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># check correlations between samples
CorMat &lt;- cor(cpm)
kable(CorMat)</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">A</th>
<th>ffected_3 H</th>
<th>ealthy_2 A</th>
<th>ffected_1 A</th>
<th>ffected_2 H</th>
<th>ealthy_1 H</th>
<th align="left">ealthy_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">Affected_3</td>
<td>1.0000000</td>
<td>0.9485111</td>
<td>0.9698672</td>
<td>0.9708302</td>
<td>0.9533155</td>
<td align="left">0.9456574</td>
</tr>
<tr class="even">
<td align="right">Healthy_2</td>
<td>0.9485111</td>
<td>1.0000000</td>
<td>0.9580786</td>
<td>0.9588884</td>
<td>0.9602608</td>
<td align="left">0.9735372</td>
</tr>
<tr class="odd">
<td align="right">Affected_1</td>
<td>0.9698672</td>
<td>0.9580786</td>
<td>1.0000000</td>
<td>0.9798980</td>
<td>0.9476967</td>
<td align="left">0.9585812</td>
</tr>
<tr class="even">
<td align="right">Affected_2</td>
<td>0.9708302</td>
<td>0.9588884</td>
<td>0.9798980</td>
<td>1.0000000</td>
<td>0.9482039</td>
<td align="left">0.9590783</td>
</tr>
<tr class="odd">
<td align="right">Healthy_1</td>
<td>0.9533155</td>
<td>0.9602608</td>
<td>0.9476967</td>
<td>0.9482039</td>
<td>1.0000000</td>
<td align="left">0.9566276</td>
</tr>
<tr class="even">
<td align="right">Healthy_3</td>
<td>0.9456574</td>
<td>0.9735372</td>
<td>0.9585812</td>
<td>0.9590783</td>
<td>0.9566276</td>
<td align="left">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Plot
heatmap.2(CorMat, trace=&quot;none&quot;)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-4-2.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Healthy_1 is sort of an outlier sample, and it is probably because it has the least reads. In my experience, this number of reads is still quite high, and the correlations are quite high, so I am not yet concerned. I think it will look like less of an outlier once we filter out the extremely lowly expressed genes that basically are just represented by the 0.5 pseudocount I added.</p>
<pre class="r"><code># Pick a cutoff for lowly expressed genes to exclude
hist(median_log2_cpm)
expr_cutoff &lt;- 2
abline(v = expr_cutoff, col = &quot;red&quot;, lwd = 3)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#How many genes pass this threshold
sum(median_log2_cpm &gt; expr_cutoff)</code></pre>
<pre><code>[1] 11730</code></pre>
<pre class="r"><code>CountTable.matrix.filtered &lt;- CountTable.matrix[median_log2_cpm &gt; expr_cutoff, ]

# Recalculate cpm from filtered count table
cpm &lt;- cpm(CountTable.matrix.filtered, log=TRUE, prior.count=0.5)

# look at correlations from filtered cpm matrix
cor(cpm) %&gt;%
heatmap.2(., trace=&quot;none&quot;, cexRow=1, cexCol=1)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-2.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Also look at similarity with MDS plot
plotMDS(cpm)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-3">
Past versions of unnamed-chunk-5-3.png
</button>
</p>
<div id="fig-unnamed-chunk-5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-3.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMDS(cpm, dim=c(3,4))</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-4">
Past versions of unnamed-chunk-5-4.png
</button>
</p>
<div id="fig-unnamed-chunk-5-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-5-4.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>samples primarily segregate by condition (healthy vs affected) as expected</p>
<p>Now start prepping data for differential expression analysis</p>
<pre class="r"><code>group &lt;- substr(colnames(CountTable.matrix.filtered), 1, 1)

# A for affected, H for healthy
group</code></pre>
<pre><code>[1] &quot;A&quot; &quot;H&quot; &quot;A&quot; &quot;A&quot; &quot;H&quot; &quot;H&quot;</code></pre>
<pre class="r"><code># y is a DGELIst object contains count table and other metadata
y &lt;- DGEList(counts = CountTable.matrix.filtered, group = group)

# calculate normalization factors using default TMM method which more accurately scales samples with different library sizes, for example while considering that highly expressed genes between conditions can have undesired effects if just using simple CountsPerMillion normalization.

y &lt;- calcNormFactors(y)
y$samples %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">gr</th>
<th>oup l</th>
<th>ib.size n</th>
<th align="left">orm.factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">Affected_3</td>
<td>A</td>
<td>27530199</td>
<td align="left">1.0138568</td>
</tr>
<tr class="even">
<td align="right">Healthy_2</td>
<td>H</td>
<td>20697090</td>
<td align="left">0.9721993</td>
</tr>
<tr class="odd">
<td align="right">Affected_1</td>
<td>A</td>
<td>29525851</td>
<td align="left">1.0289830</td>
</tr>
<tr class="even">
<td align="right">Affected_2</td>
<td>A</td>
<td>32513479</td>
<td align="left">1.0053212</td>
</tr>
<tr class="odd">
<td align="right">Healthy_1</td>
<td>H</td>
<td>17490838</td>
<td align="left">1.0187010</td>
</tr>
<tr class="even">
<td align="right">Healthy_3</td>
<td>H</td>
<td>22124071</td>
<td align="left">0.9627384</td>
</tr>
</tbody>
</table>
<p>Now estimate common, trended, and tagwise dispersion, and test for differential expression</p>
<pre class="r"><code>#estimate dispersion, common, trended, and tag-wise (a combination of gene-specific dispersion and trend)
y &lt;- estimateDisp(y)</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>sqrt(y$common.dispersion) # biological coefficient of variation</code></pre>
<pre><code>[1] 0.1426326</code></pre>
<pre class="r"><code>#Plot dispersions
plotBCV(y)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-7-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># by default, the tagwise dispersion is used for DE testing.
# Perform testing
et &lt;- exactTest(y, pair=c(&quot;H&quot;,&quot;A&quot;))
results_edgeR &lt;- topTags(et, n = nrow(CountTable.matrix.filtered), sort.by = &quot;none&quot;)

head(results_edgeR$table) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">logFC</th>
<th align="center">logCPM</th>
<th align="center">PValue</th>
<th align="center">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000279457</td>
<td align="center">-0.0019184</td>
<td align="center">2.294509</td>
<td align="center">1.0000000</td>
<td align="center">1.0000000</td>
</tr>
<tr class="even">
<td>ENSG00000248527</td>
<td align="center">-1.4384558</td>
<td align="center">3.327765</td>
<td align="center">0.0005842</td>
<td align="center">0.0028015</td>
</tr>
<tr class="odd">
<td>ENSG00000228794</td>
<td align="center">0.0824910</td>
<td align="center">3.223395</td>
<td align="center">0.6820905</td>
<td align="center">0.7909175</td>
</tr>
<tr class="even">
<td>ENSG00000230699</td>
<td align="center">0.5618857</td>
<td align="center">2.490896</td>
<td align="center">0.0208296</td>
<td align="center">0.0580221</td>
</tr>
<tr class="odd">
<td>ENSG00000188976</td>
<td align="center">-0.1738289</td>
<td align="center">6.991208</td>
<td align="center">0.2377004</td>
<td align="center">0.3809051</td>
</tr>
<tr class="even">
<td>ENSG00000187961</td>
<td align="center">-0.0084449</td>
<td align="center">4.092404</td>
<td align="center">0.9652432</td>
<td align="center">0.9806255</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#histogram of nominal P-vals
hist(results_edgeR$table$PValue)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-7-2.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># how many genes below FDR threshold
sum(results_edgeR$table$FDR &lt; .1)</code></pre>
<pre><code>[1] 4847</code></pre>
<pre class="r"><code>#MA plot showing FDR&lt;0.1
plotSmear(et, de.tags = rownames(results_edgeR)[results_edgeR$table$FDR &lt; .01])
abline(h = c(-2, 2), col = &quot;blue&quot;)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-3">
Past versions of unnamed-chunk-7-3.png
</button>
</p>
<div id="fig-unnamed-chunk-7-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-7-3.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok we did differential testing. Now let’s explore the results</p>
<p>First add common gene names to the results table</p>
<pre class="r"><code>#Use biomart to get common gene names and ensembl IDs
human_mart &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)

MyBioMartQuery &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;,&quot;external_gene_name&quot;, &quot;description&quot;), mart=human_mart, filters=&quot;ensembl_gene_id&quot;, values=rownames(results_edgeR))

results.more.descriptive &lt;- results_edgeR$table %&gt;%
  rownames_to_column(&quot;ensembl_gene_id&quot;) %&gt;%
  left_join(MyBioMartQuery, by=&quot;ensembl_gene_id&quot;)

head(results.more.descriptive) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
<th align="left">external_gene_name</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000279457</td>
<td align="right">-0.0019184</td>
<td align="right">2.294509</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
<td align="left">WASH9P</td>
<td align="left">WAS protein family homolog 9, pseudogene [Source:HGNC Symbol;Acc:HGNC:53981]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000248527</td>
<td align="right">-1.4384558</td>
<td align="right">3.327765</td>
<td align="right">0.0005842</td>
<td align="right">0.0028015</td>
<td align="left">MTATP6P1</td>
<td align="left">MT-ATP6 pseudogene 1 [Source:HGNC Symbol;Acc:HGNC:44575]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000228794</td>
<td align="right">0.0824910</td>
<td align="right">3.223395</td>
<td align="right">0.6820905</td>
<td align="right">0.7909175</td>
<td align="left">LINC01128</td>
<td align="left">long intergenic non-protein coding RNA 1128 [Source:HGNC Symbol;Acc:HGNC:49377]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000230699</td>
<td align="right">0.5618857</td>
<td align="right">2.490896</td>
<td align="right">0.0208296</td>
<td align="right">0.0580221</td>
<td align="left">AL645608.2</td>
<td align="left">novel transcript</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000188976</td>
<td align="right">-0.1738289</td>
<td align="right">6.991208</td>
<td align="right">0.2377004</td>
<td align="right">0.3809051</td>
<td align="left">NOC2L</td>
<td align="left">NOC2 like nucleolar associated transcriptional repressor [Source:HGNC Symbol;Acc:HGNC:24517]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000187961</td>
<td align="right">-0.0084449</td>
<td align="right">4.092404</td>
<td align="right">0.9652432</td>
<td align="right">0.9806255</td>
<td align="left">KLHL17</td>
<td align="left">kelch like family member 17 [Source:HGNC Symbol;Acc:HGNC:24023]</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Check if DHX38 is differentially expressed
DHX38.results &lt;- results.more.descriptive %&gt;%
  filter(external_gene_name==&quot;DHX38&quot;)

kable(DHX38.results)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
<th align="left">external_gene_name</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000140829</td>
<td align="right">-0.1891905</td>
<td align="right">7.023741</td>
<td align="right">0.1966757</td>
<td align="right">0.3311334</td>
<td align="left">DHX38</td>
<td align="left">DEAH-box helicase 38 [Source:HGNC Symbol;Acc:HGNC:17211]</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Plot smear with DHX label
PlotSmear &lt;- results.more.descriptive %&gt;%
  mutate(
    Significance=case_when(
      FDR&lt;0.1 ~ &quot;FDR&lt;0.1&quot;,
      FDR&gt;0.1 ~ &quot;Not significant&quot;,
      external_gene_name==&quot;DHX38&quot; ~ &quot;DHX38&quot;),
    Label=case_when(
      external_gene_name==&quot;DHX38&quot; ~ paste0(&quot;DHX38\nP=&quot;, signif(DHX38.results$PValue, 3)),
      !external_gene_name==&quot;DHX38&quot; ~ &quot;&quot;),
  ) %&gt;% 
  arrange(FDR)

DHX38.results &lt;- PlotSmear %&gt;%
  filter(external_gene_name==&quot;DHX38&quot;)

ggplot(PlotSmear, aes(x=logCPM, y=logFC, color=Significance, label=Label)) +
    geom_point() +
    geom_point(data=DHX38.results, color=&quot;black&quot;, show.legend = F) +
    geom_text_repel(data=DHX38.results, color=&quot;black&quot;, show.legend = F) +
    theme_bw() +
    theme(legend.title = element_blank())</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/dbbdff32af42a7bdbc1cd9c6d91adbb7bb55845f/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-8-1.png" target="_blank">dbbdff3</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok good. DHX38 is not differentially expressed. So if the alleles have some affect on splicing through DHX38, is not likely through modulating expression of DHX38 levels, but rather altering function.</p>
<p>Let’s make some plots to get to know significance testing results a little bit more… Keep in mind we don’t know much about the types of replicates these are (technical, versus biological split before differentiation, split after differentiation, etc), and that all these are from a single healthy and a single affected patient.</p>
<p>Volcano plot:</p>
<pre class="r"><code>results.more.descriptive %&gt;%
  mutate(Significance=FDR&lt;0.1) %&gt;%
  ggplot(aes(x=logFC, y=-log10(PValue), color=Significance)) +
  geom_point(alpha=0.2) +
  theme_bw()</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-9-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plot of original results of a significant gene. Useful to check polarization of the logFC values (I think I set up the <code>exactTest</code> function to be such that positive logFC means more highly expressed in affected, but let’s double check that)</p>
<pre class="r"><code>Top5DE_Genes &lt;- results.more.descriptive %&gt;%
  arrange(FDR) %&gt;% head(5) %&gt;%
  pull(ensembl_gene_id)

results.more.descriptive %&gt;%
  filter(ensembl_gene_id %in% Top5DE_Genes) %&gt;%
  dplyr::select(ensembl_gene_id, external_gene_name, logFC) %&gt;%
  kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="left">external_gene_name</th>
<th align="right">logFC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000018625</td>
<td align="left">ATP1A2</td>
<td align="right">8.652522</td>
</tr>
<tr class="even">
<td align="left">ENSG00000044524</td>
<td align="left">EPHA3</td>
<td align="right">-6.178116</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000096696</td>
<td align="left">DSP</td>
<td align="right">11.571066</td>
</tr>
<tr class="even">
<td align="left">ENSG00000196569</td>
<td align="left">LAMA2</td>
<td align="right">7.873084</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000198300</td>
<td align="left">PEG3</td>
<td align="right">13.352063</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cpm[Top5DE_Genes,] %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;ensembl_gene_id&quot;) %&gt;%
  gather(key=&quot;sample&quot;, value=&quot;log2cpm&quot;, -ensembl_gene_id) %&gt;% 
  ggplot(aes(x=sample, y=log2cpm)) +
  geom_col() +
  facet_wrap(~ensembl_gene_id) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-10-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok correct, positive FC means higher expressed in affected samples. Now let’s do gene set enrichment analysis (GSEA) to see which types of gene categories are enriched for differentially expressed genes. For this we need an ordered list of genes. We could use absolute value of log-fold change, or maybe -log10(Pvalue). Judging from the volcano plot I’d guess the results would be virtually identical either way</p>
<pre class="r"><code>#had to add very small number to Pvalue because very significant genes have Pvalues that are 0
gsea.results &lt;- results.more.descriptive %&gt;%
  mutate(RankingMetric = -log10(PValue+1E-300)) %&gt;%
  arrange(desc(RankingMetric)) %&gt;%
  dplyr::select(ensembl_gene_id, RankingMetric) %&gt;%
  deframe() %&gt;%
  gseGO(ont=&quot;ALL&quot;, OrgDb=org.Hs.eg.db, keyType=&#39;ENSEMBL&#39;, nPerm=500000)</code></pre>
<p>Now let’s explore the GSEA results</p>
<pre class="r"><code>gsea.results.df &lt;- as.data.frame(gsea.results)
gsea.results.df %&gt;% 
  group_by(ONTOLOGY) %&gt;%
  filter(rank(p.adjust, ties.method=&quot;first&quot;)&lt;=8) %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  facet_grid(ONTOLOGY~., scales = &quot;free&quot;) +
  scale_colour_viridis_c(trans=&quot;log10&quot;, limits=c(2E-4, 1), direction=-1, option=&quot;D&quot;) +
  xlab(&quot;Enrichment&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 60, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/dbbdff32af42a7bdbc1cd9c6d91adbb7bb55845f/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-12-1.png" target="_blank">dbbdff3</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/a0c762759c7f066dc175ebf7988a53401ad16cc8/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-12-1.png" target="_blank">a0c7627</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-12-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gsea.results.df %&gt;%
  arrange(p.adjust) %&gt;% head(10) %&gt;%
  dplyr::select(ID, Description, ONTOLOGY, p.adjust, setSize, enrichmentScore)</code></pre>
<pre><code>           ID                                    Description ONTOLOGY
1  GO:0000904 cell morphogenesis involved in differentiation       BP
2  GO:0002009                 morphogenesis of an epithelium       BP
3  GO:0005887          integral component of plasma membrane       CC
4  GO:0007268                 chemical synaptic transmission       BP
5  GO:0007389                  pattern specification process       BP
6  GO:0007409                                   axonogenesis       BP
7  GO:0007420                              brain development       BP
8  GO:0007423                      sensory organ development       BP
9  GO:0009986                                   cell surface       CC
10 GO:0031226         intrinsic component of plasma membrane       CC
       p.adjust setSize enrichmentScore
1  0.0005502598     466       0.7726367
2  0.0005502598     342       0.7776557
3  0.0005502598     445       0.7737560
4  0.0005502598     347       0.7832492
5  0.0005502598     233       0.7959597
6  0.0005502598     288       0.8064676
7  0.0005502598     494       0.7412182
8  0.0005502598     298       0.7866344
9  0.0005502598     290       0.7825923
10 0.0005502598     481       0.7704378</code></pre>
<pre class="r"><code>gsea.results.df %&gt;%
  arrange(desc(enrichmentScore)) %&gt;% head(10) %&gt;%
  dplyr::select(ID, Description, ONTOLOGY, p.adjust, setSize)</code></pre>
<pre><code>           ID
1  GO:0044065
2  GO:0031420
3  GO:0098533
4  GO:0010881
5  GO:0010882
6  GO:0043576
7  GO:0014808
8  GO:1903514
9  GO:0048701
10 GO:0086009
                                                                                        Description
1                                                          regulation of respiratory system process
2                                                                          alkali metal ion binding
3                                                  ATPase dependent transmembrane transport complex
4  regulation of cardiac muscle contraction by regulation of the release of sequestered calcium ion
5                                 regulation of cardiac muscle contraction by calcium ion signaling
6                                                        regulation of respiratory gaseous exchange
7                         release of sequestered calcium ion into cytosol by sarcoplasmic reticulum
8                          release of sequestered calcium ion into cytosol by endoplasmic reticulum
9                                                          embryonic cranial skeleton morphogenesis
10                                                                          membrane repolarization
   ONTOLOGY   p.adjust setSize
1        BP 0.01587967      12
2        MF 0.03714877      10
3        CC 0.04422252      10
4        BP 0.03776860      14
5        BP 0.02269330      17
6        BP 0.03920944      15
7        BP 0.03619861      19
8        BP 0.03619861      19
9        BP 0.02127983      23
10       BP 0.03137866      23</code></pre>
<p>Lot’s of interesting stuff. Let’s write out the full table of results. Here is a plot of a few terms that catch’s the eye (<a href="GO:0007409" class="uri">GO:0007409</a>, axonogenesis). Though, another term that might point towards the interpretation of these differences being more related to the ipsc differentiation differences with n=1 (or maybe n=3; I still don’t fully understand the experiment design).</p>
<p>This plot captures the essence of how GSEA works. A rug depicting instances of genes belonging to sets is shown on x-axis. Some gene sets are enriched for genes at the top of the ordered list, leading to more enrichment than you expect by chance as you move from the top (left) to bottom (right) of the ordered list.</p>
<pre class="r"><code>gseaplot2(gsea.results, geneSetID = c(&quot;GO:0007409&quot;, &quot;GO:0007268&quot;, &quot;GO:0000904&quot;, &quot;GO:0086009&quot;), pvalue_table = F)</code></pre>
<p><img src="figure/20191203_DiffExpression.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DiffExpression.Rmd/unnamed-chunk-13-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Another thing to keep in mind with this and any GSEA is that there is a lot of overlap in some gene set terms, as is clear above in the rug plot along the x-axis. Some sets are huge (hundreds of terms).</p>
<pre class="r"><code>#Write out gsea results
write.table(gsea.results.df, file=&quot;../output/GeneExpression.gsea.tab&quot;, quote=F, sep=&#39;\t&#39;, row.names = F)

#write differential expression results
write.table(results.more.descriptive, file=&quot;../output/GeneExpression.DE.tab&quot;, quote=F, sep=&#39;\t&#39;, row.names = F)</code></pre>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>Perhaps some neuronal related gene sets enriched, though with small sample sizes and confounding DHX38 genotype with other individual genetic differences in this study design, it is difficult to draw strong conclusions about the effect of these DHX38 mutations.</p>
<p>At the very least we can be sure DHX38 is not very differently expresesed.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] enrichplot_1.4.0       org.Hs.eg.db_3.8.2     AnnotationDbi_1.46.1  
 [4] IRanges_2.18.3         S4Vectors_0.22.1       Biobase_2.44.0        
 [7] BiocGenerics_0.30.0    clusterProfiler_3.12.0 ggrepel_0.8.1         
[10] biomaRt_2.40.5         gplots_3.0.1.1         corrplot_0.84         
[13] edgeR_3.26.8           limma_3.40.6           knitr_1.26            
[16] forcats_0.4.0          stringr_1.4.0          dplyr_0.8.3           
[19] purrr_0.3.3            readr_1.3.1            tidyr_1.0.0           
[22] tibble_2.1.3           ggplot2_3.2.1          tidyverse_1.3.0       

loaded via a namespace (and not attached):
  [1] readxl_1.3.1        backports_1.1.5     fastmatch_1.1-0    
  [4] workflowr_1.5.0     plyr_1.8.5          igraph_1.2.4.2     
  [7] lazyeval_0.2.2      splines_3.6.1       BiocParallel_1.18.1
 [10] urltools_1.7.3      digest_0.6.23       htmltools_0.4.0    
 [13] GOSemSim_2.10.0     viridis_0.5.1       GO.db_3.8.2        
 [16] gdata_2.18.0        fansi_0.4.0         magrittr_1.5       
 [19] memoise_1.1.0       graphlayouts_0.5.0  modelr_0.1.5       
 [22] prettyunits_1.0.2   colorspace_1.4-1    blob_1.2.0         
 [25] rvest_0.3.5         haven_2.2.0         xfun_0.11          
 [28] crayon_1.3.4        RCurl_1.95-4.12     jsonlite_1.6       
 [31] zeallot_0.1.0       glue_1.3.1          polyclip_1.10-0    
 [34] gtable_0.3.0        UpSetR_1.4.0        scales_1.1.0       
 [37] DOSE_3.10.2         DBI_1.0.0           Rcpp_1.0.2         
 [40] viridisLite_0.3.0   progress_1.2.2      gridGraphics_0.4-1 
 [43] bit_1.1-14          europepmc_0.3       httr_1.4.1         
 [46] fgsea_1.10.1        RColorBrewer_1.1-2  ellipsis_0.3.0     
 [49] pkgconfig_2.0.3     XML_3.98-1.20       farver_2.0.1       
 [52] dbplyr_1.4.2        locfit_1.5-9.1      ggplotify_0.0.4    
 [55] tidyselect_0.2.5    labeling_0.3        rlang_0.4.1        
 [58] reshape2_1.4.3      later_1.0.0         munsell_0.5.0      
 [61] cellranger_1.1.0    tools_3.6.1         cli_2.0.0          
 [64] generics_0.0.2      RSQLite_2.1.4       broom_0.5.2        
 [67] ggridges_0.5.1      evaluate_0.14       yaml_2.2.0         
 [70] bit64_0.9-7         fs_1.3.1            tidygraph_1.1.2    
 [73] caTools_1.17.1.3    ggraph_2.0.0        nlme_3.1-143       
 [76] whisker_0.4         DO.db_2.9           xml2_1.2.2         
 [79] compiler_3.6.1      rstudioapi_0.10     curl_4.3           
 [82] reprex_0.3.0        tweenr_1.0.1        stringi_1.4.3      
 [85] highr_0.8           lattice_0.20-38     Matrix_1.2-18      
 [88] vctrs_0.2.0         pillar_1.4.2        lifecycle_0.1.0    
 [91] BiocManager_1.30.10 triebeard_0.3.0     data.table_1.12.8  
 [94] cowplot_1.0.0       bitops_1.0-6        httpuv_1.5.2       
 [97] qvalue_2.16.0       R6_2.4.1            promises_1.1.0     
[100] KernSmooth_2.23-16  gridExtra_2.3       MASS_7.3-51.4      
[103] gtools_3.8.1        assertthat_0.2.1    rprojroot_1.3-2    
[106] withr_2.1.2         hms_0.5.2           grid_3.6.1         
[109] rmarkdown_1.18      rvcheck_0.1.7       git2r_0.26.1       
[112] ggforce_0.3.1       lubridate_1.7.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
