<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2020-01-14" />

<title>Untitled</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Untitled</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">January 14, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rna-seq-dhx38/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree2ce85b8d6032575c1844103c3da6463192639abetargetblank2ce85b8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/rna-seq-dhx38/tree/2ce85b8d6032575c1844103c3da6463192639abe" target="_blank">2ce85b8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree2ce85b8d6032575c1844103c3da6463192639abetargetblank2ce85b8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/20200110_DiffExpression_MoreReplicates.Rmd
    Untracked:  output/GeneExpression.DE.WithMoreReplicates.tab
    Untracked:  output/GeneExpression.gsea.WithMoreReplicates.tab

Unstaged changes:
    Modified:   analysis/20190105_DifferentialSplicingMoreReplicates.Rmd
    Modified:   analysis/20191203_DiffExpression.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/analysis/20190105_DifferentialSplicingMoreReplicates.Rmd" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
<td>
add splicing analysis with more reps
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/rna-seq-dhx38/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/20190105_DifferentialSplicingMoreReplicates.html" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
<td>
add splicing analysis with more reps
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Exploratory differential expression analysis comparing RNA-seq from iPSC derived neurons from healthy control vs patient with biallelic mutations in DHX38 that may affect splicing and gene expression. Some things I want to check is what kinds of gene sets are differentially spliced, as well as if there are certain intron features associated with differentially spliced introns (weaker 5’ss motif perhaps)</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>I did the same leafcutter differential splicing analysis as before but this time with more replicates, wherein the biological replicates were combined at the level of fastq files.</p>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)
library(psych)
library(corrplot)
library(ggrepel)</code></pre>
<p>Now let’s establish that the 1 healthy vs 1 control (where 3 healthy bioligical replicates are combined into 1, same with control) is comparable with the previous analysis where the replicates are left separate. Obviously I expect the 3vs3 to have more inflated P-values, but I want to check that the P-values are still well correlated.</p>
<pre class="r"><code>sig_3v3 &lt;- read.table(&quot;../output/leafcutter_ds_analyses/original_cluster_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(sig_3v3) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cluster</th>
<th align="left">status</th>
<th align="right">loglr</th>
<th align="right">df</th>
<th align="right">p</th>
<th align="right">p.adjust</th>
<th align="left">genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:clu_10533_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_10534_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">KLHL17</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_10535_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">KLHL17</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_10536_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">ISG15</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_10537_+</td>
<td align="left">Not enough valid samples</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">ISG15</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_10538_+</td>
<td align="left">Success</td>
<td align="right">1.859755</td>
<td align="right">1</td>
<td align="right">0.0537801</td>
<td align="right">0.2106285</td>
<td align="left">AGRN</td>
</tr>
</tbody>
</table>
<pre class="r"><code>sig_1v1 &lt;- read.table(&quot;../output/leafcutter_ds_analyses/originalCombined_cluster_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(sig_1v1) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cluster</th>
<th align="left">status</th>
<th align="right">loglr</th>
<th align="right">df</th>
<th align="right">p</th>
<th align="right">p.adjust</th>
<th align="left">genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:clu_10533_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_10534_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">KLHL17</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_10535_+</td>
<td align="left">Success</td>
<td align="right">0.0703501</td>
<td align="right">1</td>
<td align="right">0.7075860</td>
<td align="right">0.8695491</td>
<td align="left">KLHL17</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_10536_+</td>
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">ISG15</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_10537_+</td>
<td align="left">Success</td>
<td align="right">2.6978256</td>
<td align="right">1</td>
<td align="right">0.0201870</td>
<td align="right">0.2417123</td>
<td align="left">ISG15</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_10538_+</td>
<td align="left">Success</td>
<td align="right">1.4471068</td>
<td align="right">1</td>
<td align="right">0.0888982</td>
<td align="right">0.3455686</td>
<td align="left">AGRN</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Check that the dimensions of the two results tables are identical... they should be
dim(sig_3v3)</code></pre>
<pre><code>[1] 14690     7</code></pre>
<pre class="r"><code>dim(sig_1v1)</code></pre>
<pre><code>[1] 14690     7</code></pre>
<pre class="r"><code>#Check that the P-values are well correlated
cor.test(sig_3v3$p, sig_1v1$p)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  sig_3v3$p and sig_1v1$p
t = 505.42, df = 8438, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9831855 0.9845505
sample estimates:
      cor 
0.9838823 </code></pre>
<pre class="r"><code>#Visualize this correlation with scatter plot
left_join(sig_3v3, sig_1v1, by=&quot;cluster&quot;, suffix=c(&quot;.3v3&quot;, &quot;.1v1&quot;)) %&gt;%
  ggplot(aes(x=-log10(p.3v3), y=-log10(p.1v1))) +
  geom_point(alpha=0.2) +
  theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-2-1.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok. Yes, it seems you get well correlated P-values. I feel justified in combining replicates as I have and running leafcutter’s differential splicing analysis on those combined replicates. This is a much simpler approach than dealing with these different levels of replication with a mixed model for example. Let’s check how many significant clusters you get this way…</p>
<pre class="r"><code>hist(sig_3v3$p)</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-3-1.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>hist(sig_1v1$p)</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-3-2.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>table(sig_1v1$p.adjust&lt;0.1)</code></pre>
<pre><code>
FALSE  TRUE 
13237   139 </code></pre>
<pre class="r"><code>table(sig_3v3$p.adjust&lt;0.1)</code></pre>
<pre><code>
FALSE  TRUE 
 7280  1160 </code></pre>
<p>Good. there are still some significant clusters to work with, but as expected, there are much less. Now we can add more replicates (different lines, each with multiple replicates similarly combined at the level of fastq files) and check how many significant clusters we get.</p>
<pre class="r"><code>sig &lt;- read.table(&quot;../output/leafcutter_ds_analyses/AdditionalReps_cluster_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)
table(sig$p.adjust&lt;0.1)</code></pre>
<pre><code>
FALSE  TRUE 
13881    24 </code></pre>
<pre class="r"><code>table(sig$p.adjust&lt;0.2)</code></pre>
<pre><code>
FALSE  TRUE 
13813    92 </code></pre>
<p>Ok. only 24 significant differentially spliced clusters at FDR&lt;0.1. These are likely more robustly associated with the DHX sample of interest carrying biallelic mutations. However, let’s permute the contrast samples as a way to assess how much of the inflated P-values might be due to natural genetic variation between unrelated samples and other technical effects that are unrelated to the DHX38 sample specifically. If there is a similar amount of P-value inflation, I will conclude that most of the differentially spliced clusters are still not specific to an additional DHX38 effect, but rather, represent the natural amount of genetic and biological/technical effects present in any of these lines.</p>
<pre class="r"><code>#Read in the significant results from the permutated data
Permutation1ClusterSig &lt;- read.table(&quot;../output/leafcutter_ds_analyses/AdditionalRepsPerm1_cluster_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

Permutation2ClusterSig &lt;- read.table(&quot;../output/leafcutter_ds_analyses/AdditionalRepsPerm2_cluster_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

Permutation3ClusterSig &lt;- read.table(&quot;../output/leafcutter_ds_analyses/AdditionalRepsPerm3_cluster_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

data.frame(RealResults=sig$p,
           Permutation1Results=Permutation1ClusterSig$p,
           Permutation2Results=Permutation2ClusterSig$p,
           Permutation3Results=Permutation3ClusterSig$p) %&gt;% gather(value=&quot;Actual.P&quot;) %&gt;%
  group_by(key) %&gt;%
  mutate(Expected.P=percent_rank(Actual.P)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(color=key, x=-log10(Expected.P), y=-log10(Actual.P))) +
  geom_point() +
  geom_abline() +
  scale_color_manual(labels = c(Permutation1Results=&quot;Permuted; [BIH4] v [DHX, XM001, JG]&quot;,
                                Permutation2Results=&quot;Permuted; [XM001] v [DHX, BIH4, JG]&quot;,
                                Permutation3Results=&quot;Permuted; [JG] v [DHX, XM001, BIH4]&quot;,
                                RealResults=&quot;Actual; [DHX] v [BIH4, XM001, JG]&quot;),
                     values=c(&quot;red&quot;, &quot;blue&quot;, &quot;purple&quot;, &quot;black&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-5-1.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, similar to the same plot made for gene expression analysis, there is not drastically more P-value inflation than in some of the permutations, and in fact there is more small P-value inflation in one of the permutations. I conclude that the effect of the DHX38 mutation is quite small, relative to the effect of comparing any random sample of different unrelated lines. Most of the significantly different splice events are probably not due to DHX38. Nonetheless, assuming that there really is a DHX38-specific splicing defect, the splice events we do find are the best prioritized list of DHX38-dependednt splice events that we can do, with the data at hand.</p>
<p>Also, with only 24 significantly differentially spliced clusters, there isn’t much reasonable downstream analysis we can do with regards to looking for intron features that explain these. Let’s make a volcano plot to visualize, labelling the genes with the significantly changed splicing events.</p>
<pre class="r"><code>effect_sizes &lt;- read.table(&quot;../output/leafcutter_ds_analyses/AdditionalReps_effect_sizes.txt.gz&quot;, sep=&#39;\t&#39;, header=T)


#how many clusters tested (&quot;Success&quot;)
table(sig$status) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">&lt;=1 sample with coverage&gt;0</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">&lt;=1 sample with coverage&gt;min_coverage</td>
<td align="right">722</td>
</tr>
<tr class="odd">
<td align="left">Not enough valid samples</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">Success</td>
<td align="right">13905</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#how many introns for each cluster
effect_sizes %&gt;%
  mutate(Cluster=sub(&quot;.+?:(clu_.+?_.+?)&quot;,&quot;\\1&quot;,intron)) %&gt;%
  group_by(Cluster) %&gt;%
  tally() %&gt;%
  ggplot(aes(x=n)) +
  geom_histogram() +
  scale_x_continuous(breaks = seq(1, 10, by = 1), limits=c(1,10)) +
  xlab(&quot;Num introns per cluster&quot;) +
  ylab(&quot;Freq&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#histogram of Pvalues
hist(sig$p)</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#how many significant (p.adjust &lt; 0.1)
table(sig$p.adjust&lt;0.1) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">13881</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">24</td>
</tr>
</tbody>
</table>
<pre class="r"><code># merge significance and cluster tables
leafcutter.merged &lt;- effect_sizes %&gt;%
  mutate(cluster=gsub(&quot;(.+?:).+?:.+?:(clu.+?)&quot;, &quot;\\1\\2&quot;, intron, perl=T)) %&gt;%
  mutate(junc_id=gsub(&quot;(.+?:.+?:.+?):clu.+&quot;, &quot;\\1&quot;, intron, perl=T)) %&gt;%
  left_join(sig, by=&quot;cluster&quot;)

# volcano plot of cluster-pvalues and largest within-cluster-delta-psi
leafcutter.merged %&gt;%
  mutate(abs.deltapsi = abs(deltapsi)) %&gt;%
  group_by(cluster) %&gt;% 
  top_n(n=1, abs.deltapsi) %&gt;%
  mutate(sig=p.adjust&lt;0.1 &amp; abs.deltapsi&gt;0.1) %&gt;%
  mutate(label=dplyr::case_when(
    sig==TRUE ~ as.character(genes),
    sig==FALSE ~ &quot;&quot;
  )) %&gt;%
  mutate(alpha=case_when(
    sig==TRUE ~ 1,
    sig==FALSE ~ 0.05
  )) %&gt;%
  ggplot(aes(x=deltapsi, y=-log10(p), color=sig, label=label)) +
    geom_point(aes(alpha=alpha)) +
    geom_text_repel(size=3) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /> Let’s check for intron clusters that overlap DHX38 gene, in case the gene itself is differentially spliced, possibly as a result of the mutations acting in cis.</p>
<p>There are no clusters over DHX38. Note that leafcutter output only lists clusters that have more than 1 intron (not simple constitutive splicing) and that pass some default filters for minimum coverage over each the splice junction.</p>
<pre class="r"><code>#Filter based on leafcutter&#39;s gene annotations.
sig %&gt;% dplyr::filter(grepl(&quot;DHX38&quot;,genes))</code></pre>
<pre><code>           cluster  status   loglr df          p  p.adjust genes
1 chr16:clu_6060_+ Success 2.86251  1 0.01672485 0.3738497 DHX38</code></pre>
<pre class="r"><code>#Filter also based on genome coordinates of DHX38
effect_sizes %&gt;%
  separate(intron, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;cluster&quot;), sep=&quot;:&quot;) %&gt;%
  filter(chrom==&quot;chr16&quot; &amp; start&gt;= 72093613 &amp; stop &lt;= 72112912)</code></pre>
<pre><code>  chrom    start     stop    cluster      logef         H          A
1 chr16 72106104 72107340 clu_6060_+ -0.6478399 0.1210321 0.03632065
2 chr16 72106117 72107340 clu_6060_+  0.6478399 0.8789679 0.96367935
     deltapsi
1 -0.08471147
2  0.08471147</code></pre>
<p>what kinds of intron features distinguish differentially spliced introns, which may be more enlightening about general splicing mechanisms due to this DHX38 genotype.</p>
<p>I have scored the 5’ss and 3’ss motifs for each intron tested using a position specific score matrix based on annotated introns. I additionally used <a href="http://regulatorygenomics.upf.edu/Software/SVM_BP/">svm-bpfinder</a> to score the best computationally predicted branch for each intron. I also annotated whether each intron is annotated, an unanottated alt3’ss, an unanotated alt5’ss, an unannotated new pairing of splice sites, or new intron (unannotated 5’ and unannotated 3’ss). All of that work is hidden in the <code>code</code> section and output into a file in <code>output</code>.</p>
<p>Let’s see if any of those features correlate with differentially expressed introns. If there is the DHX38 mutations preferentially activates less consensus splice sites for example, I expect the directionality to be consistent. Therefore, I will correlate those scores with the logeff (polarized effect size) for each intron, for significant introns.</p>
<pre class="r"><code>#read intron features
IntronFeatures &lt;- read.table(&quot;../output/IntronFeatures.txt.gz&quot;, sep=&#39;\t&#39;, header=T) %&gt;%
  separate(Intron, c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;), sep = &quot;_&quot;) %&gt;%
  mutate(newStop=as.numeric(stop)+1,
         newStart=as.numeric(start)) %&gt;%
  mutate(
    junc_id=paste(paste0(&quot;chr&quot;, chrom), newStart, newStop, sep=&quot;:&quot;),
    IntronLength=newStop-newStart) %&gt;%
  dplyr::select(-c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;newStop&quot;, &quot;newStart&quot;))

head(IntronFeatures) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">strand</th>
<th align="left">type</th>
<th align="left">donor_seq</th>
<th align="right">DonorScore</th>
<th align="left">acceptor_seq</th>
<th align="right">AcceptorScore</th>
<th align="right">eqez</th>
<th align="right">ss_dist</th>
<th align="left">bp_seq</th>
<th align="right">bp_scr</th>
<th align="right">y_cont</th>
<th align="right">ppt_off</th>
<th align="right">ppt_len</th>
<th align="right">ppt_scr</th>
<th align="right">svm_scr</th>
<th align="left">junc_id</th>
<th align="right">IntronLength</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">+</td>
<td align="left">NewIntron</td>
<td align="left">CAGGTGAAG</td>
<td align="right">7.509</td>
<td align="left">GCTGTGGTCTTCATCTGCAGGTG</td>
<td align="right">9.120</td>
<td align="right">29</td>
<td align="right">22</td>
<td align="left">tcttgatgc</td>
<td align="right">0.7736716</td>
<td align="right">0.7058824</td>
<td align="right">9</td>
<td align="right">9</td>
<td align="right">19</td>
<td align="right">0.5400024</td>
<td align="left">chr1:11671:12010</td>
<td align="right">339</td>
</tr>
<tr class="even">
<td align="left">+</td>
<td align="left">NewIntron</td>
<td align="left">CGGGTATCA</td>
<td align="right">2.853</td>
<td align="left">GCTGTGGTCTTCATCTGCAGGTG</td>
<td align="right">9.120</td>
<td align="right">29</td>
<td align="right">22</td>
<td align="left">tcttgatgc</td>
<td align="right">0.7736716</td>
<td align="right">0.7058824</td>
<td align="right">9</td>
<td align="right">9</td>
<td align="right">19</td>
<td align="right">0.5400024</td>
<td align="left">chr1:11844:12010</td>
<td align="right">166</td>
</tr>
<tr class="odd">
<td align="left">+</td>
<td align="left">AnnotatedSpliceSite</td>
<td align="left">GAGGTGAGA</td>
<td align="right">9.498</td>
<td align="left">TAGCCTCTGTTCCCACGAAGGCA</td>
<td align="right">4.753</td>
<td align="right">17</td>
<td align="right">19</td>
<td align="left">gccttagcc</td>
<td align="right">-0.9155434</td>
<td align="right">0.7857143</td>
<td align="right">2</td>
<td align="right">13</td>
<td align="right">22</td>
<td align="right">0.3754140</td>
<td align="left">chr1:12721:13221</td>
<td align="right">500</td>
</tr>
<tr class="even">
<td align="left">+</td>
<td align="left">NewIntron</td>
<td align="left">AAAGTGAGG</td>
<td align="right">7.351</td>
<td align="left">AGAAGGGGAGAAGAGGAAAGGGG</td>
<td align="right">-14.250</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">chr1:13114:183635</td>
<td align="right">170521</td>
</tr>
<tr class="odd">
<td align="left">-</td>
<td align="left">NewIntron</td>
<td align="left">AGGGTAGGA</td>
<td align="right">6.363</td>
<td align="left">CATGACACCTTCCCCCCCAGACC</td>
<td align="right">6.602</td>
<td align="right">27</td>
<td align="right">16</td>
<td align="left">ccatgacac</td>
<td align="right">1.3904305</td>
<td align="right">0.9090909</td>
<td align="right">1</td>
<td align="right">11</td>
<td align="right">20</td>
<td align="right">1.3628254</td>
<td align="left">chr1:14668:14970</td>
<td align="right">302</td>
</tr>
<tr class="even">
<td align="left">-</td>
<td align="left">NewIntron</td>
<td align="left">CTTGTAATT</td>
<td align="right">3.934</td>
<td align="left">ACTCAGCCTTTTCCCTCCAGGCA</td>
<td align="right">11.310</td>
<td align="right">14</td>
<td align="right">20</td>
<td align="left">ccctcactc</td>
<td align="right">2.5121196</td>
<td align="right">0.8666667</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">28</td>
<td align="right">1.8628447</td>
<td align="left">chr1:14829:14930</td>
<td align="right">101</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Check an example of a good donor score and a bad donor score
#Good splice donor (positions -3 to +6)
IntronFeatures %&gt;%
  top_n(1, DonorScore) %&gt;% head(1) %&gt;%
  dplyr::select(donor_seq, DonorScore)</code></pre>
<pre><code>  donor_seq DonorScore
1 CAGGTAAGT       12.6</code></pre>
<pre class="r"><code>#Bad splice donor
IntronFeatures %&gt;%
  top_n(-1, DonorScore) %&gt;% head(1) %&gt;%
  dplyr::select(donor_seq, DonorScore)</code></pre>
<pre><code>  donor_seq DonorScore
1 TTCATTCCA     -15.08</code></pre>
<pre class="r"><code># Check that the number of types of unannotated splicing is similar in both strands. Sometimes bugs related to coordinates produce wonky results
IntronFeatures %&gt;%
  filter(strand==&quot;-&quot;) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
              35029               26917               21176              112893 
          NewIntron 
              37099 </code></pre>
<pre class="r"><code>IntronFeatures %&gt;%
  filter(strand==&quot;+&quot;) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
              35787               27336               21710              115391 
          NewIntron 
              38410 </code></pre>
<pre class="r"><code># merge the intron features with the leafcutter results and filter for the significantly differentially spliced introns that have the greatest effect size within cluster
leafcutter.merged.w.features &lt;- leafcutter.merged %&gt;%
  mutate(abs.effectsize = abs(logef)) %&gt;%
  group_by(cluster) %&gt;% 
  top_n(n=1, abs.effectsize) %&gt;%
  ungroup() %&gt;%
  filter(p.adjust&lt;0.2) %&gt;%
  left_join(IntronFeatures, by=&quot;junc_id&quot;)

# Are specific types of unannotated splicing more prevalent?
leafcutter.merged.w.features %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
                  1                   7                   8                  72 
          NewIntron 
                  6 </code></pre>
<pre class="r"><code>colnames(leafcutter.merged.w.features)</code></pre>
<pre><code> [1] &quot;intron&quot;         &quot;logef&quot;          &quot;H&quot;              &quot;A&quot;             
 [5] &quot;deltapsi&quot;       &quot;cluster&quot;        &quot;junc_id&quot;        &quot;status&quot;        
 [9] &quot;loglr&quot;          &quot;df&quot;             &quot;p&quot;              &quot;p.adjust&quot;      
[13] &quot;genes&quot;          &quot;abs.effectsize&quot; &quot;strand&quot;         &quot;type&quot;          
[17] &quot;donor_seq&quot;      &quot;DonorScore&quot;     &quot;acceptor_seq&quot;   &quot;AcceptorScore&quot; 
[21] &quot;eqez&quot;           &quot;ss_dist&quot;        &quot;bp_seq&quot;         &quot;bp_scr&quot;        
[25] &quot;y_cont&quot;         &quot;ppt_off&quot;        &quot;ppt_len&quot;        &quot;ppt_scr&quot;       
[29] &quot;svm_scr&quot;        &quot;IntronLength&quot;  </code></pre>
<pre class="r"><code># Set up features and effect sizes into a cleaner data frame
IntronFeaturesAndEffectSizes &lt;- leafcutter.merged.w.features %&gt;%
  dplyr::select(junc_id, logef, DonorScore, AcceptorScore, ss_dist, bp_scr, ppt_len, ppt_scr, svm_scr, IntronLength) %&gt;%
  column_to_rownames(&quot;junc_id&quot;)

# See if there is any correlations
CorrelationsWithFeatures &lt;- corr.test(x=IntronFeaturesAndEffectSizes[,1],
                         y=IntronFeaturesAndEffectSizes[,-1],
                         method=&quot;spearman&quot;,
                         adjust=&quot;none&quot;)

# table of spearmans r, pvalue, and n
data.frame(Spearman.r= t(CorrelationsWithFeatures$r), p=t(CorrelationsWithFeatures$p), n=t(CorrelationsWithFeatures$n)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">S</th>
<th align="left">pearman.r</th>
<th align="center">p</th>
<th align="left">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">DonorScore</td>
<td align="left">0.0214669</td>
<td align="center">0.8372843</td>
<td align="left">94</td>
</tr>
<tr class="even">
<td align="right">AcceptorScore</td>
<td align="left">0.0247519</td>
<td align="center">0.8128087</td>
<td align="left">94</td>
</tr>
<tr class="odd">
<td align="right">ss_dist</td>
<td align="left">-0.0822056</td>
<td align="center">0.4490769</td>
<td align="left">87</td>
</tr>
<tr class="even">
<td align="right">bp_scr</td>
<td align="left">0.0297989</td>
<td align="center">0.7840949</td>
<td align="left">87</td>
</tr>
<tr class="odd">
<td align="right">ppt_len</td>
<td align="left">-0.0668364</td>
<td align="center">0.5385021</td>
<td align="left">87</td>
</tr>
<tr class="even">
<td align="right">ppt_scr</td>
<td align="left">-0.1043431</td>
<td align="center">0.3361503</td>
<td align="left">87</td>
</tr>
<tr class="odd">
<td align="right">svm_scr</td>
<td align="left">-0.0401693</td>
<td align="center">0.7118263</td>
<td align="left">87</td>
</tr>
<tr class="even">
<td align="right">IntronLength</td>
<td align="left">0.1171120</td>
<td align="center">0.2609604</td>
<td align="left">94</td>
</tr>
</tbody>
</table>
<p>None of the features tested seem to correlate with the effect size of largest-effect introns in significant clusters.</p>
<p>Check if the significant positive effect-size splicing changes have more cryptic 5’ss, or more cryptic 3’ss than negative effect-size splicing changes.</p>
<p>When doing this analysis, some people think in terms of deltapsi, some people might think in terms of a different metric of effect size. Let’s look at both.</p>
<pre class="r"><code># leafcutter&#39;s logef effect size, versus deltapsi
qplot(leafcutter.merged.w.features$logef, leafcutter.merged.w.features$deltapsi) + theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#significant positive deltapsi introns
leafcutter.merged.w.features %&gt;%
  filter(deltapsi &gt; 0) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
                  1                   1                   4                  49 
          NewIntron 
                  5 </code></pre>
<pre class="r"><code>#significant negative deltapsi introns
leafcutter.merged.w.features %&gt;%
  filter(deltapsi &lt; 0) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
                  0                   6                   4                  23 
          NewIntron 
                  1 </code></pre>
<pre class="r"><code>leafcutter.merged.w.features %&gt;%
  ggplot(aes(x=deltapsi, color=type)) +
    stat_ecdf(geom = &quot;step&quot;) +
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>There simply aren’t many significant introns to do this kind of analyses. Nonetheless, the polarization in effect sizse for annotated introns and unanottated cryptic 5’ss seems to be the most likely candidates to be significant. Let’s see if these effects are significant or could just be chance.</p>
<pre class="r"><code>## are alt annotated splice sites differentially regulated?
wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;AnnotatedSpliceSite&quot;) %&gt;% pull(deltapsi))</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;AnnotatedSpliceSite&quot;) %&gt;%     pull(deltapsi)
V = 1643.5, p-value = 0.06486
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code># same test but using logef instead of deltapsi
wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;AnnotatedSpliceSite&quot;) %&gt;% pull(logef))</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;AnnotatedSpliceSite&quot;) %&gt;%     pull(logef)
V = 1679.5, p-value = 0.04053
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code>## are cryptic unannotated 5&#39;ss differentially regulated?
wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;Alt5ss&quot;) %&gt;% pull(deltapsi))</code></pre>
<pre><code>
    Wilcoxon signed rank test

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;Alt5ss&quot;) %&gt;%     pull(deltapsi)
V = 5, p-value = 0.1563
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code># same test but using logef instead of deltapsi
wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;Alt5ss&quot;) %&gt;% pull(logef))</code></pre>
<pre><code>
    Wilcoxon signed rank test

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;Alt5ss&quot;) %&gt;%     pull(logef)
V = 3, p-value = 0.07813
alternative hypothesis: true location is not equal to 0</code></pre>
<p>These effects aren’t signicant enough to warrant mention in my opinion.</p>
<p>Note that thus far, all my alternative splicing type classifications are based on identifying splice donors and splice acceptors that aren’t in annotated transcripts (Ensembl annotations)… In other words, alternative 5’ss events that are annotated (most identified alternative splicing is annotated), are simply classified as “annotated”. But if the effect about downregulation of cryptic 5’ss is real, I also expect the minor intron for annotated alternative 5’ss to decrease in the affected. So let’s try to classify splicing types regardless of annotations, at least for clusters with only two or three introns where this is relatively straightforward:</p>
<p>For example, for a cluster with two introns, if the splice donor coordinates is listed twice in the dataset and the splice acceptor coordinates is listed once, we know it is alternative 5’ss usage.</p>
<pre class="r"><code>#Reclassify splicing types
leafcutter.merged.AS.reclassified &lt;- leafcutter.merged %&gt;% 
  dplyr::select(intron) %&gt;%
  separate(intron, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;cluster&quot;), sep=&quot;:&quot;, remove=F) %&gt;%
  mutate(strand=sub(&quot;clu_\\d+_(.+)&quot;, &quot;\\1&quot;, cluster)) %&gt;%
  mutate(
    SpliceDonorCoord=case_when(
      strand==&quot;+&quot; ~ paste(chrom, start, strand),
      strand==&quot;-&quot; ~ paste(chrom, stop, strand)),
    SpliceAcceptorCoord=case_when(
      strand==&quot;+&quot; ~ paste(chrom, stop, strand),
      strand==&quot;-&quot; ~ paste(chrom, start, strand))) %&gt;%
  dplyr::select(intron, cluster, SpliceDonorCoord,SpliceAcceptorCoord) %&gt;%
  add_count(cluster, name=&quot;IntronsPerCluster&quot;) %&gt;%
  add_count(SpliceDonorCoord, name=&quot;SpliceDonorCount&quot;) %&gt;%
  add_count(SpliceAcceptorCoord, name=&quot;SpliceAcceptorCount&quot;) %&gt;%
  mutate(SplicingType=case_when(
    IntronsPerCluster==2 &amp; SpliceDonorCount==2 ~ &quot;Alt3ss&quot;,
    IntronsPerCluster==2 &amp; SpliceAcceptorCount==2 ~ &quot;Alt5ss&quot;,
    IntronsPerCluster==3 &amp; SpliceDonorCount==2 &amp; SpliceAcceptorCount==2 ~ &quot;CassetteExonSkipped&quot;,
    IntronsPerCluster==3 &amp; SpliceDonorCount==1 &amp; SpliceAcceptorCount==2 ~ &quot;CassetteExonIncluded.DownstreamIntron&quot;,
    IntronsPerCluster==3 &amp; SpliceDonorCount==2 &amp; SpliceAcceptorCount==1 ~ &quot;CassetteExonIncluded.UpstreamIntron&quot;,
    IntronsPerCluster==3 &amp; SpliceDonorCount==3 &amp; SpliceAcceptorCount==1 ~ &quot;Alt3ss&quot;,
    IntronsPerCluster==3 &amp; SpliceDonorCount==1 &amp; SpliceAcceptorCount==3 ~ &quot;Alt5ss&quot;,
    IntronsPerCluster&gt;=4 ~ &quot;Complex&quot;
  )) %&gt;%
  dplyr::select(intron, SplicingType) %&gt;%
  left_join(leafcutter.merged, by=&quot;intron&quot;)

#Number of splicing types
table(leafcutter.merged.AS.reclassified$SplicingType) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Alt3ss</td>
<td align="right">7713</td>
</tr>
<tr class="even">
<td align="left">Alt5ss</td>
<td align="right">6208</td>
</tr>
<tr class="odd">
<td align="left">CassetteExonIncluded.DownstreamIntron</td>
<td align="right">4748</td>
</tr>
<tr class="even">
<td align="left">CassetteExonIncluded.UpstreamIntron</td>
<td align="right">4748</td>
</tr>
<tr class="odd">
<td align="left">CassetteExonSkipped</td>
<td align="right">4748</td>
</tr>
<tr class="even">
<td align="left">Complex</td>
<td align="right">12892</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Check that the classifications make sense by manually inspecting some intron coordinates for a few clusters
head(leafcutter.merged.AS.reclassified) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">intron</th>
<th align="left">SplicingType</th>
<th align="right">logef</th>
<th align="right">H</th>
<th align="right">A</th>
<th align="right">deltapsi</th>
<th align="left">cluster</th>
<th align="left">junc_id</th>
<th align="left">status</th>
<th align="right">loglr</th>
<th align="right">df</th>
<th align="right">p</th>
<th align="right">p.adjust</th>
<th align="left">genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:829104:847654:clu_10533_+</td>
<td align="left">CassetteExonIncluded.UpstreamIntron</td>
<td align="right">-0.1365485</td>
<td align="right">0.4032932</td>
<td align="right">0.3549527</td>
<td align="right">-0.0483405</td>
<td align="left">chr1:clu_10533_+</td>
<td align="left">chr1:829104:847654</td>
<td align="left">Success</td>
<td align="right">0.0942184</td>
<td align="right">2</td>
<td align="right">0.9100840</td>
<td align="right">0.9746209</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">chr1:829104:851927:clu_10533_+</td>
<td align="left">CassetteExonSkipped</td>
<td align="right">0.0657430</td>
<td align="right">0.2097756</td>
<td align="right">0.2260261</td>
<td align="right">0.0162505</td>
<td align="left">chr1:clu_10533_+</td>
<td align="left">chr1:829104:851927</td>
<td align="left">Success</td>
<td align="right">0.0942184</td>
<td align="right">2</td>
<td align="right">0.9100840</td>
<td align="right">0.9746209</td>
<td align="left">LINC01128</td>
</tr>
<tr class="odd">
<td align="left">chr1:847806:851927:clu_10533_+</td>
<td align="left">CassetteExonIncluded.DownstreamIntron</td>
<td align="right">0.0708055</td>
<td align="right">0.3869312</td>
<td align="right">0.4190212</td>
<td align="right">0.0320900</td>
<td align="left">chr1:clu_10533_+</td>
<td align="left">chr1:847806:851927</td>
<td align="left">Success</td>
<td align="right">0.0942184</td>
<td align="right">2</td>
<td align="right">0.9100840</td>
<td align="right">0.9746209</td>
<td align="left">LINC01128</td>
</tr>
<tr class="even">
<td align="left">chr1:964167:964349:clu_10535_+</td>
<td align="left">Alt5ss</td>
<td align="right">0.1486869</td>
<td align="right">0.6494433</td>
<td align="right">0.7138112</td>
<td align="right">0.0643679</td>
<td align="left">chr1:clu_10535_+</td>
<td align="left">chr1:964167:964349</td>
<td align="left">Success</td>
<td align="right">0.2020411</td>
<td align="right">1</td>
<td align="right">0.5249885</td>
<td align="right">0.8322711</td>
<td align="left">KLHL17</td>
</tr>
<tr class="odd">
<td align="left">chr1:964180:964349:clu_10535_+</td>
<td align="left">Alt5ss</td>
<td align="right">-0.1486869</td>
<td align="right">0.3505567</td>
<td align="right">0.2861888</td>
<td align="right">-0.0643679</td>
<td align="left">chr1:clu_10535_+</td>
<td align="left">chr1:964180:964349</td>
<td align="left">Success</td>
<td align="right">0.2020411</td>
<td align="right">1</td>
<td align="right">0.5249885</td>
<td align="right">0.8322711</td>
<td align="left">KLHL17</td>
</tr>
<tr class="even">
<td align="left">chr1:1008279:1013984:clu_10537_+</td>
<td align="left">Alt5ss</td>
<td align="right">-0.5311327</td>
<td align="right">0.1826391</td>
<td align="right">0.0717020</td>
<td align="right">-0.1109371</td>
<td align="left">chr1:clu_10537_+</td>
<td align="left">chr1:1008279:1013984</td>
<td align="left">Success</td>
<td align="right">1.2342663</td>
<td align="right">1</td>
<td align="right">0.1161463</td>
<td align="right">0.5755373</td>
<td align="left">ISG15</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Check that the classifications make sense in that the fraction of junction reads for introns upstream of cassette exons should strongly positively correlate with the fraction for introns downstream of cassette exons
PlotExonSkippingSpliceRatios &lt;- leafcutter.merged.AS.reclassified %&gt;%
  dplyr::select(cluster, SplicingType, H) %&gt;%
  filter(SplicingType %in% c(&quot;CassetteExonIncluded.DownstreamIntron&quot;, &quot;CassetteExonIncluded.UpstreamIntron&quot;, &quot;CassetteExonSkipped&quot;)) %&gt;%
  spread(key=&quot;SplicingType&quot;, value=&quot;H&quot;)

ggplot(PlotExonSkippingSpliceRatios, aes(x=CassetteExonIncluded.DownstreamIntron, y=CassetteExonIncluded.UpstreamIntron)) +
    geom_point()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-1.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#And there should generally be anti correlation with the exon skipped isoform and either of the exon included introns
ggplot(PlotExonSkippingSpliceRatios, aes(x=CassetteExonIncluded.DownstreamIntron, y=CassetteExonSkipped)) +
    geom_point()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-2.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#And finally, cassette skipped exons should have weaker splice sites.
leafcutter.merged.AS.reclassified %&gt;%
  left_join(IntronFeatures, by=&quot;junc_id&quot;) %&gt;%
  filter(SplicingType %in% c(&quot;CassetteExonIncluded.DownstreamIntron&quot;, &quot;CassetteExonIncluded.UpstreamIntron&quot;)) %&gt;%
  dplyr::select(SplicingType, DonorScore, AcceptorScore) %&gt;%
  gather(key=&quot;Motif&quot;, value=&quot;MotifScore&quot;, -SplicingType) %&gt;%
  mutate(Motif=case_when(
    Motif==&quot;DonorScore&quot; ~ &quot;5&#39;ss&quot;,
    Motif== &quot;AcceptorScore&quot; ~ &quot;3&#39;ss&quot;)) %&gt;%
  ggplot(aes(x=MotifScore, color=Motif, linetype=SplicingType)) +
    stat_ecdf(geom = &quot;step&quot;) +
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-3">
Past versions of unnamed-chunk-11-3.png
</button>
</p>
<div id="fig-unnamed-chunk-11-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-3.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>A&lt;-leafcutter.merged.AS.reclassified %&gt;%
  left_join(IntronFeatures, by=&quot;junc_id&quot;) %&gt;%
  filter(SplicingType %in% c(&quot;CassetteExonIncluded.DownstreamIntron&quot;, &quot;CassetteExonIncluded.UpstreamIntron&quot;)) %&gt;%
  dplyr::select(SplicingType, DonorScore, AcceptorScore)
wilcox.test(DonorScore~SplicingType, data=A, alternative=&quot;less&quot;)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  DonorScore by SplicingType
W = 9365248, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is less than 0</code></pre>
<pre class="r"><code>wilcox.test(AcceptorScore~SplicingType, data=A)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  AcceptorScore by SplicingType
W = 13367826, p-value &lt; 2.2e-16
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>leafcutter.merged.AS.reclassified %&gt;%
  left_join(IntronFeatures, by=&quot;junc_id&quot;) %&gt;%
  filter(SplicingType %in% c(&quot;CassetteExonIncluded.DownstreamIntron&quot;, &quot;CassetteExonIncluded.UpstreamIntron&quot;)) %&gt;%
  dplyr::select(SplicingType, svm_scr) %&gt;%
  ggplot(aes(x=svm_scr, linetype=SplicingType)) +
    stat_ecdf(geom = &quot;step&quot;) +
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-4">
Past versions of unnamed-chunk-11-4.png
</button>
</p>
<div id="fig-unnamed-chunk-11-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/4e8e59a820dd4e9f6d9300a3e5560def09567b0a/docs/figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-11-4.png" target="_blank">4e8e59a</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2020-01-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Okay yes, for cassette exons, upstream introns have weaker splice acceptors than down stream splice acceptors, downstream introns have weaker splice donors than upstream splice donors. That pattern isn’t apparent for branchpoints though, so that signal may be too weak, or predictions on branchpoints may be lousy.</p>
<p>But all in all it seems that my classification of alternative splicing types makes sense. Now let’s check if the minor isoform (as assessed in healthy samples) for each splicing type is has a general polarity to effect size. Specifically, if alt5’ss have negative effect sizes.</p>
<pre class="r"><code>MinorIsoforms &lt;- leafcutter.merged.AS.reclassified %&gt;%
  filter(p.adjust&lt;0.3) %&gt;%
  group_by(cluster) %&gt;%
  top_n(-1, H) %&gt;%
  ungroup()

#histogram of splicing fraction of minor isoforms (intron with smallest splicing fraction within cluster)
hist(MinorIsoforms$H)</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>MajorIsoforms &lt;- leafcutter.merged.AS.reclassified %&gt;%
  filter(p.adjust&lt;0.3) %&gt;%
  group_by(cluster) %&gt;%
  top_n(1, H) %&gt;%
  ungroup()

#histogram of splicing fraction of major isoforms (intron with largest splicing fraction within cluster)
hist(MajorIsoforms$H)</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Now make plots to get to know if minor isoform alt5&#39;ss have generally gone down. Same for alt 3&#39;ss.
MinorIsoforms %&gt;%
  filter(SplicingType %in% c(&quot;Alt5ss&quot;, &quot;Alt3ss&quot;)) %&gt;%
ggplot(aes(x=deltapsi, color=SplicingType)) +
    stat_ecdf(geom = &quot;step&quot;) +
    xlim(c(-1,1))+
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Mirrored plot for major isoforms.
MajorIsoforms %&gt;%
    filter(SplicingType %in% c(&quot;Alt5ss&quot;, &quot;Alt3ss&quot;)) %&gt;%
ggplot(aes(x=deltapsi, color=SplicingType)) +
    stat_ecdf(geom = &quot;step&quot;) +
    xlim(c(-1,1))+
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Is this significant?
wilcox.test(
  MinorIsoforms %&gt;%
  filter(SplicingType==&quot;Alt5ss&quot;) %&gt;% pull(logef))</code></pre>
<pre><code>
    Wilcoxon signed rank test

data:  MinorIsoforms %&gt;% filter(SplicingType == &quot;Alt5ss&quot;) %&gt;% pull(logef)
V = 146, p-value = 0.4678
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(
  MinorIsoforms %&gt;%
  filter(SplicingType==&quot;Alt3ss&quot;) %&gt;% pull(logef))</code></pre>
<pre><code>
    Wilcoxon signed rank test

data:  MinorIsoforms %&gt;% filter(SplicingType == &quot;Alt3ss&quot;) %&gt;% pull(logef)
V = 192, p-value = 0.594
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code>#Try exon skipping
leafcutter.merged.AS.reclassified %&gt;%
  filter(p.adjust&lt;0.3) %&gt;%
  filter(SplicingType %in% c(&quot;CassetteExonSkipped&quot;, &quot;CassetteExonIncluded.DownstreamIntron&quot;, &quot;CassetteExonIncluded.UpstreamIntron&quot;)) %&gt;%
  ggplot(aes(x=deltapsi, color=SplicingType)) +
    stat_ecdf(geom = &quot;step&quot;) +
    xlim(c(-1,1))+
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20190105_DifferentialSplicingMoreReplicates.Rmd/unnamed-chunk-12-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>So there is no apparent difference in alt5’ss or alt3’ss usage when I considered all easily classifiable introns, regardless of annotation. In other words, the minor isoforms for 5’ss were equally up and down regulated. This is mirrored in when looking at major isoforms.</p>
<p>Finally, since all of this type of analysis didn’t return anything very interesting with confidence, I should browse the raw aligned read data in IGV to explore patterns that I might not have picked up on this differential splicing analysis. To aid in IGV exploration, let’s write out a bed file of the most significant clusters so we can quickly jump to the regions of interest that are differentially spliced</p>
<pre class="r"><code>#bed file, with chromosome, start, stop, name, score, strand, extra info

BedFile &lt;- leafcutter.merged.AS.reclassified %&gt;%
  dplyr::select(intron, SplicingType, logef, deltapsi, p) %&gt;%
  separate(intron, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;cluster&quot;), remove=F) %&gt;%
  mutate(extra=paste(logef, deltapsi, SplicingType),
         score=-log10(p),
         strand=&quot;.&quot;,
         name=paste0(chrom,&quot;:&quot;,start,&quot;-&quot;,stop)
         )%&gt;%
  dplyr::select(chrom, start, stop, name, score, strand, extra )

head(BedFile) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">chrom</th>
<th align="left">start</th>
<th align="left">stop</th>
<th align="left">name</th>
<th align="right">score</th>
<th align="left">strand</th>
<th align="left">extra</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1</td>
<td align="left">829104</td>
<td align="left">847654</td>
<td align="left">chr1:829104-847654</td>
<td align="right">0.0409185</td>
<td align="left">.</td>
<td align="left">-0.136548512778748 -0.0483405254670429 CassetteExonIncluded.UpstreamIntron</td>
</tr>
<tr class="even">
<td align="left">chr1</td>
<td align="left">829104</td>
<td align="left">851927</td>
<td align="left">chr1:829104-851927</td>
<td align="right">0.0409185</td>
<td align="left">.</td>
<td align="left">0.0657429995098606 0.0162505111995736 CassetteExonSkipped</td>
</tr>
<tr class="odd">
<td align="left">chr1</td>
<td align="left">847806</td>
<td align="left">851927</td>
<td align="left">chr1:847806-851927</td>
<td align="right">0.0409185</td>
<td align="left">.</td>
<td align="left">0.0708055132688876 0.0320900142674692 CassetteExonIncluded.DownstreamIntron</td>
</tr>
<tr class="even">
<td align="left">chr1</td>
<td align="left">964167</td>
<td align="left">964349</td>
<td align="left">chr1:964167-964349</td>
<td align="right">0.2798502</td>
<td align="left">.</td>
<td align="left">0.14868691020407 0.0643679254899464 Alt5ss</td>
</tr>
<tr class="odd">
<td align="left">chr1</td>
<td align="left">964180</td>
<td align="left">964349</td>
<td align="left">chr1:964180-964349</td>
<td align="right">0.2798502</td>
<td align="left">.</td>
<td align="left">-0.14868691020407 -0.0643679254899463 Alt5ss</td>
</tr>
<tr class="even">
<td align="left">chr1</td>
<td align="left">1008279</td>
<td align="left">1013984</td>
<td align="left">chr1:1008279-1013984</td>
<td align="right">0.9349945</td>
<td align="left">.</td>
<td align="left">-0.531132748000042 -0.110937105555015 Alt5ss</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.table(BedFile, file=&quot;../output/DiffSplicingRegions.WithMoreReplicates.bed&quot;, sep=&#39;\t&#39;, row.names = F, col.names = F, quote=F)</code></pre>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggrepel_0.8.1          corrplot_0.84          psych_1.8.12          
 [4] enrichplot_1.4.0       org.Hs.eg.db_3.8.2     AnnotationDbi_1.46.1  
 [7] IRanges_2.18.3         S4Vectors_0.22.1       Biobase_2.44.0        
[10] BiocGenerics_0.30.0    clusterProfiler_3.12.0 knitr_1.26            
[13] forcats_0.4.0          stringr_1.4.0          dplyr_0.8.3           
[16] purrr_0.3.3            readr_1.3.1            tidyr_1.0.0           
[19] tibble_2.1.3           ggplot2_3.2.1          tidyverse_1.3.0       

loaded via a namespace (and not attached):
  [1] fgsea_1.10.1        colorspace_1.4-1    ellipsis_0.3.0     
  [4] ggridges_0.5.1      rprojroot_1.3-2     qvalue_2.16.0      
  [7] fs_1.3.1            rstudioapi_0.10     farver_2.0.1       
 [10] urltools_1.7.3      graphlayouts_0.5.0  bit64_0.9-7        
 [13] fansi_0.4.0         lubridate_1.7.4     xml2_1.2.2         
 [16] splines_3.6.1       mnormt_1.5-5        GOSemSim_2.10.0    
 [19] polyclip_1.10-0     zeallot_0.1.0       jsonlite_1.6       
 [22] workflowr_1.5.0     broom_0.5.2         GO.db_3.8.2        
 [25] dbplyr_1.4.2        ggforce_0.3.1       BiocManager_1.30.10
 [28] compiler_3.6.1      httr_1.4.1          rvcheck_0.1.7      
 [31] backports_1.1.5     assertthat_0.2.1    Matrix_1.2-18      
 [34] lazyeval_0.2.2      cli_2.0.0           later_1.0.0        
 [37] tweenr_1.0.1        htmltools_0.4.0     prettyunits_1.0.2  
 [40] tools_3.6.1         igraph_1.2.4.2      gtable_0.3.0       
 [43] glue_1.3.1          reshape2_1.4.3      DO.db_2.9          
 [46] fastmatch_1.1-0     Rcpp_1.0.2          cellranger_1.1.0   
 [49] vctrs_0.2.0         nlme_3.1-143        ggraph_2.0.0       
 [52] xfun_0.11           rvest_0.3.5         lifecycle_0.1.0    
 [55] DOSE_3.10.2         europepmc_0.3       MASS_7.3-51.4      
 [58] scales_1.1.0        tidygraph_1.1.2     hms_0.5.2          
 [61] promises_1.1.0      RColorBrewer_1.1-2  yaml_2.2.0         
 [64] memoise_1.1.0       gridExtra_2.3       UpSetR_1.4.0       
 [67] triebeard_0.3.0     stringi_1.4.3       RSQLite_2.1.4      
 [70] highr_0.8           BiocParallel_1.18.1 rlang_0.4.1        
 [73] pkgconfig_2.0.3     evaluate_0.14       lattice_0.20-38    
 [76] labeling_0.3        cowplot_1.0.0       bit_1.1-14         
 [79] tidyselect_0.2.5    plyr_1.8.5          magrittr_1.5       
 [82] R6_2.4.1            generics_0.0.2      DBI_1.0.0          
 [85] foreign_0.8-72      pillar_1.4.2        haven_2.2.0        
 [88] whisker_0.4         withr_2.1.2         modelr_0.1.5       
 [91] crayon_1.3.4        rmarkdown_1.18      viridis_0.5.1      
 [94] progress_1.2.2      grid_3.6.1          readxl_1.3.1       
 [97] data.table_1.12.8   blob_1.2.0          git2r_0.26.1       
[100] reprex_0.3.0        digest_0.6.23       httpuv_1.5.2       
[103] gridGraphics_0.4-1  munsell_0.5.0       ggplotify_0.0.4    
[106] viridisLite_0.3.0  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
