<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2019-12-03" />

<title>Untitled</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Untitled</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">December 03, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-12-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rna-seq-dhx38/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree0f4f9122fee8e7f4727e1ebae9fbcaf8084d46c8targetblank0f4f912a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/rna-seq-dhx38/tree/0f4f9122fee8e7f4727e1ebae9fbcaf8084d46c8" target="_blank">0f4f912</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree0f4f9122fee8e7f4727e1ebae9fbcaf8084d46c8targetblank0f4f912a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/20191203_DiffExpression.Rmd
    Untracked:  analysis/20191203_DifferentialSplicing.Rmd
    Untracked:  output/GeneExpression.DE.tab
    Untracked:  output/GeneExpression.gsea.tab

Unstaged changes:
    Modified:   _workflowr.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Exploratory differential expression analysis comparing RNA-seq from iPSC derived neurons from healthy control vs patient with biallelic mutations in DHX38 that may affect splicing and gene expression. Some things I want to check is what kinds of gene sets are differentially spliced, as well as if there are certain intron features associated with differentially spliced introns (weaker 5’ss motif perhaps)</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Raw RNA-seq data has been aligned and collapsed into leafcutter splice junction count table (rows of introns, columns of samples, each cell is exonic read counts) using code in <code>code</code> section of this repo. Also, leafcutter has been run to identify differentially spliced clusters of splice junctions, creating output files in <code>output</code>. Here I will explore that output and maybe some other exploratory analyses.</p>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(edgeR)
library(biomaRt)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)</code></pre>
<p>load in leafcutter output</p>
<pre class="r"><code>sig &lt;- read.table(&quot;../output/differential_splicing_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(sig) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cluster</th>
<th align="left">status</th>
<th align="right">loglr</th>
<th align="right">df</th>
<th align="right">p</th>
<th align="right">p.adjust</th>
<th align="left">genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:clu_1000_-</td>
<td align="left">Success</td>
<td align="right">1.9730317</td>
<td align="right">1</td>
<td align="right">0.0469812</td>
<td align="right">0.1628228</td>
<td align="left">GNB1</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_1001_-</td>
<td align="left">Success</td>
<td align="right">0.2795714</td>
<td align="right">2</td>
<td align="right">0.7561078</td>
<td align="right">0.8609193</td>
<td align="left">GNB1</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_1002_-</td>
<td align="left">Success</td>
<td align="right">4.1659670</td>
<td align="right">3</td>
<td align="right">0.0396273</td>
<td align="right">0.1455953</td>
<td align="left">GNB1</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_1003_-</td>
<td align="left">Success</td>
<td align="right">0.0850228</td>
<td align="right">1</td>
<td align="right">0.6800713</td>
<td align="right">0.8148571</td>
<td align="left">FAAP20</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_1004_-</td>
<td align="left">Not enough valid samples</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">MORN1</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_1005_-</td>
<td align="left">Not enough valid samples</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">PANK4</td>
</tr>
</tbody>
</table>
<pre class="r"><code>effect_sizes &lt;- read.table(&quot;../output/differential_splicing_effect_sizes.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(effect_sizes) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">intron</th>
<th align="right">logef</th>
<th align="right">H</th>
<th align="right">A</th>
<th align="right">deltapsi</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:1787053:1787322:clu_1000_-</td>
<td align="right">0.2136016</td>
<td align="right">0.9383983</td>
<td align="right">0.9589358</td>
<td align="right">0.0205375</td>
</tr>
<tr class="even">
<td align="left">chr1:1787053:1787326:clu_1000_-</td>
<td align="right">-0.2136016</td>
<td align="right">0.0616017</td>
<td align="right">0.0410642</td>
<td align="right">-0.0205375</td>
</tr>
<tr class="odd">
<td align="left">chr1:1815862:1817837:clu_1001_-</td>
<td align="right">-0.0975622</td>
<td align="right">0.4873418</td>
<td align="right">0.4832734</td>
<td align="right">-0.0040684</td>
</tr>
<tr class="even">
<td align="left">chr1:1815862:1825397:clu_1001_-</td>
<td align="right">0.1824135</td>
<td align="right">0.0059958</td>
<td align="right">0.0078669</td>
<td align="right">0.0018710</td>
</tr>
<tr class="odd">
<td align="left">chr1:1817875:1825397:clu_1001_-</td>
<td align="right">-0.0848513</td>
<td align="right">0.5066623</td>
<td align="right">0.5088598</td>
<td align="right">0.0021974</td>
</tr>
<tr class="even">
<td align="left">chr1:1825499:1839190:clu_1002_-</td>
<td align="right">0.0350052</td>
<td align="right">0.5886054</td>
<td align="right">0.6221619</td>
<td align="right">0.0335565</td>
</tr>
</tbody>
</table>
<p>Get a sense of number of intron clusters tested, number significant, etc</p>
<pre class="r"><code>#how many clusters tested (&quot;Success&quot;)
table(sig$status)</code></pre>
<pre><code>
&lt;=1 sample with coverage&gt;min_coverage              Not enough valid samples 
                                  212                                  1278 
                              Success 
                                 3619 </code></pre>
<pre class="r"><code>#histogram of Pvalues
hist(sig$p)</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#how many significant (p.adjust &lt; 0.1)
table(sig$p.adjust&lt;0.1)</code></pre>
<pre><code>
FALSE  TRUE 
 2860   759 </code></pre>
<pre class="r"><code># merge significance and cluster tables
leafcutter.merged &lt;- effect_sizes %&gt;%
  mutate(cluster=gsub(&quot;(.+?:).+?:.+?:(clu.+?)&quot;, &quot;\\1\\2&quot;, intron, perl=T)) %&gt;%
  mutate(junc_id=gsub(&quot;(.+?:.+?:.+?):clu.+&quot;, &quot;\\1&quot;, intron, perl=T)) %&gt;%
  left_join(sig, by=&quot;cluster&quot;)

# volcano plot of cluster-pvalues and largest within-cluster-delta-psi
leafcutter.merged %&gt;%
  mutate(abs.deltapsi = abs(deltapsi)) %&gt;%
  group_by(cluster) %&gt;% 
  top_n(n=1, abs.deltapsi) %&gt;%
  mutate(sig=p.adjust&lt;0.05) %&gt;%
  ggplot(aes(x=deltapsi, y=-log10(p), color=sig)) +
    geom_point(alpha=0.05) +
    theme_bw()</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>let’s do a gsea to see what genes have differentially spliced clusters of introns. For this, we need an ordered gene list, with each gene appearing once. Since the biological question related to this is what are the genes or gene sets that are getting disrupted to cause the organismal phenotype, i think it is important to consider not just significance (P-value) but also effect size, perhaps as measured by delta-psi. Since significane and effect size can be quite different in a splicing analysis (see volcano plot above) because of low read counts on some junctions, let’s first filter out all non-significant splicing changes (to filter out the noisy effect size estimates) Then let’s make an ordered list based on the maximum delta-PSI for each gene.</p>
<pre class="r"><code>#let&#39;s use a fairly permissive significance threshold to be more inclusive. how about FDR&gt;0.3
hist(leafcutter.merged$p.adjust)</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#filter and get max delta-psi for each gene. Keep in mind we need to filter out non-valid genes for the clusters or introns that have more than one gene listed in the genes column
OrderedGeneList &lt;- leafcutter.merged %&gt;%
  filter(p.adjust&lt;0.3) %&gt;%
  mutate(abs.deltapsi = abs(deltapsi)) %&gt;%
  group_by(genes) %&gt;% 
  top_n(n=1, abs.deltapsi) %&gt;%
  filter(!grepl(&quot;,&quot;,genes)) %&gt;%
  arrange(desc(abs.deltapsi)) %&gt;%
  dplyr::select(genes, abs.deltapsi) %&gt;%
  deframe()

head(OrderedGeneList)</code></pre>
<pre><code>    PQBP1    FIGNL1    KANSL1    KANSL1    RNF220     PARP2 
0.9807278 0.6522834 0.5597076 0.5597076 0.5443519 0.5362952 </code></pre>
<pre class="r"><code>gsea.results &lt;- gseGO(OrderedGeneList, ont=&quot;ALL&quot;, OrgDb=org.Hs.eg.db, keyType=&#39;SYMBOL&#39;, nPerm=100000)

gsea.results.df &lt;- as.data.frame(gsea.results)

dim(gsea.results.df)</code></pre>
<pre><code>[1] 0 9</code></pre>
<p>No significant GSEA results at the default P.adjust&lt;0.5 threshold.</p>
<p>Ok, besides identifying the genes (or kinds of genes) through which the disease phenotype is potentially mediated, maybe the other questions I should be asking is what kinds of intron features distinguish differentially spliced introns, which may be more enlightening about general splicing mechanisms in these DHX38 mutants.</p>
</div>
<div id="to-do" class="section level2">
<h2>TO DO</h2>
<ul>
<li><p>what types of alternative splicing are more differentially spliced (alt 5’ss, exon-skipping, etc)</p></li>
<li><p>splice site motif analysis</p></li>
</ul>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] enrichplot_1.4.0       org.Hs.eg.db_3.8.2     AnnotationDbi_1.46.1  
 [4] IRanges_2.18.3         S4Vectors_0.22.1       Biobase_2.44.0        
 [7] BiocGenerics_0.30.0    clusterProfiler_3.12.0 biomaRt_2.40.5        
[10] edgeR_3.26.8           limma_3.40.6           knitr_1.26            
[13] forcats_0.4.0          stringr_1.4.0          dplyr_0.8.3           
[16] purrr_0.3.3            readr_1.3.1            tidyr_1.0.0           
[19] tibble_2.1.3           ggplot2_3.2.1          tidyverse_1.2.1       

loaded via a namespace (and not attached):
  [1] fgsea_1.10.1        colorspace_1.4-1    ggridges_0.5.1     
  [4] rprojroot_1.3-2     qvalue_2.16.0       fs_1.3.1           
  [7] rstudioapi_0.10     farver_1.1.0        urltools_1.7.3     
 [10] graphlayouts_0.5.0  ggrepel_0.8.1       bit64_0.9-7        
 [13] lubridate_1.7.4     xml2_1.2.2          splines_3.6.1      
 [16] GOSemSim_2.10.0     polyclip_1.10-0     zeallot_0.1.0      
 [19] jsonlite_1.6        workflowr_1.5.0     broom_0.5.2        
 [22] GO.db_3.8.2         ggforce_0.3.1       BiocManager_1.30.9 
 [25] compiler_3.6.1      httr_1.4.1          rvcheck_0.1.6      
 [28] backports_1.1.5     assertthat_0.2.1    Matrix_1.2-17      
 [31] lazyeval_0.2.2      cli_1.1.0           later_1.0.0        
 [34] tweenr_1.0.1        htmltools_0.4.0     prettyunits_1.0.2  
 [37] tools_3.6.1         igraph_1.2.4.1      gtable_0.3.0       
 [40] glue_1.3.1          reshape2_1.4.3      DO.db_2.9          
 [43] fastmatch_1.1-0     Rcpp_1.0.2          cellranger_1.1.0   
 [46] vctrs_0.2.0         nlme_3.1-142        ggraph_2.0.0       
 [49] xfun_0.11           rvest_0.3.5         lifecycle_0.1.0    
 [52] XML_3.98-1.20       DOSE_3.10.2         europepmc_0.3      
 [55] MASS_7.3-51.4       scales_1.0.0        tidygraph_1.1.2    
 [58] hms_0.5.2           promises_1.1.0      RColorBrewer_1.1-2 
 [61] yaml_2.2.0          memoise_1.1.0       gridExtra_2.3      
 [64] UpSetR_1.4.0        triebeard_0.3.0     stringi_1.4.3      
 [67] RSQLite_2.1.2       highr_0.8           BiocParallel_1.18.1
 [70] rlang_0.4.1         pkgconfig_2.0.3     bitops_1.0-6       
 [73] evaluate_0.14       lattice_0.20-38     labeling_0.3       
 [76] cowplot_1.0.0       bit_1.1-14          tidyselect_0.2.5   
 [79] plyr_1.8.4          magrittr_1.5        R6_2.4.1           
 [82] generics_0.0.2      DBI_1.0.0           pillar_1.4.2       
 [85] haven_2.2.0         withr_2.1.2         RCurl_1.95-4.12    
 [88] modelr_0.1.5        crayon_1.3.4        rmarkdown_1.17     
 [91] viridis_0.5.1       progress_1.2.2      locfit_1.5-9.1     
 [94] grid_3.6.1          readxl_1.3.1        data.table_1.12.6  
 [97] blob_1.2.0          git2r_0.26.1        digest_0.6.22      
[100] httpuv_1.5.2        gridGraphics_0.4-1  munsell_0.5.0      
[103] ggplotify_0.0.4     viridisLite_0.3.0  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
