<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2019-12-04" />

<title>Untitled</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Untitled</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">December 04, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-12-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rna-seq-dhx38/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree0d7521b0051dd8fd31ed5857d788357da9fdd302targetblank0d7521ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/rna-seq-dhx38/tree/0d7521b0051dd8fd31ed5857d788357da9fdd302" target="_blank">0d7521b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree0d7521b0051dd8fd31ed5857d788357da9fdd302targetblank0d7521ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/20191203_DifferentialSplicing.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/0d7521b0051dd8fd31ed5857d788357da9fdd302/analysis/20191203_DifferentialSplicing.Rmd" target="_blank">0d7521b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/rna-seq-dhx38/0d7521b0051dd8fd31ed5857d788357da9fdd302/docs/20191203_DifferentialSplicing.html" target="_blank">0d7521b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/analysis/20191203_DifferentialSplicing.Rmd" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
<td>
initialize site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/rna-seq-dhx38/c1d660db6d38073970096cd71db8fa190693c0d4/docs/20191203_DifferentialSplicing.html" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
<td>
initialize site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Exploratory differential expression analysis comparing RNA-seq from iPSC derived neurons from healthy control vs patient with biallelic mutations in DHX38 that may affect splicing and gene expression. Some things I want to check is what kinds of gene sets are differentially spliced, as well as if there are certain intron features associated with differentially spliced introns (weaker 5’ss motif perhaps)</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Raw RNA-seq data has been aligned and collapsed into leafcutter splice junction count table (rows of introns, columns of samples, each cell is exonic read counts) using code in <code>code</code> section of this repo. Also, leafcutter has been run to identify differentially spliced clusters of splice junctions, creating output files in <code>output</code>. Here I will explore that output and maybe some other exploratory analyses.</p>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(edgeR)
library(biomaRt)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)
library(psych)
library(corrplot)</code></pre>
<p>load in leafcutter output</p>
<pre class="r"><code>sig &lt;- read.table(&quot;../output/differential_splicing_significance.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(sig) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cluster</th>
<th align="left">status</th>
<th align="right">loglr</th>
<th align="right">df</th>
<th align="right">p</th>
<th align="right">p.adjust</th>
<th align="left">genes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:clu_1000_-</td>
<td align="left">Success</td>
<td align="right">1.9730317</td>
<td align="right">1</td>
<td align="right">0.0469812</td>
<td align="right">0.1628228</td>
<td align="left">GNB1</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_1001_-</td>
<td align="left">Success</td>
<td align="right">0.2795714</td>
<td align="right">2</td>
<td align="right">0.7561078</td>
<td align="right">0.8609193</td>
<td align="left">GNB1</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_1002_-</td>
<td align="left">Success</td>
<td align="right">4.1659670</td>
<td align="right">3</td>
<td align="right">0.0396273</td>
<td align="right">0.1455953</td>
<td align="left">GNB1</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_1003_-</td>
<td align="left">Success</td>
<td align="right">0.0850228</td>
<td align="right">1</td>
<td align="right">0.6800713</td>
<td align="right">0.8148571</td>
<td align="left">FAAP20</td>
</tr>
<tr class="odd">
<td align="left">chr1:clu_1004_-</td>
<td align="left">Not enough valid samples</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">MORN1</td>
</tr>
<tr class="even">
<td align="left">chr1:clu_1005_-</td>
<td align="left">Not enough valid samples</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">PANK4</td>
</tr>
</tbody>
</table>
<pre class="r"><code>effect_sizes &lt;- read.table(&quot;../output/differential_splicing_effect_sizes.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(effect_sizes) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">intron</th>
<th align="right">logef</th>
<th align="right">H</th>
<th align="right">A</th>
<th align="right">deltapsi</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1:1787053:1787322:clu_1000_-</td>
<td align="right">0.2136016</td>
<td align="right">0.9383983</td>
<td align="right">0.9589358</td>
<td align="right">0.0205375</td>
</tr>
<tr class="even">
<td align="left">chr1:1787053:1787326:clu_1000_-</td>
<td align="right">-0.2136016</td>
<td align="right">0.0616017</td>
<td align="right">0.0410642</td>
<td align="right">-0.0205375</td>
</tr>
<tr class="odd">
<td align="left">chr1:1815862:1817837:clu_1001_-</td>
<td align="right">-0.0975622</td>
<td align="right">0.4873418</td>
<td align="right">0.4832734</td>
<td align="right">-0.0040684</td>
</tr>
<tr class="even">
<td align="left">chr1:1815862:1825397:clu_1001_-</td>
<td align="right">0.1824135</td>
<td align="right">0.0059958</td>
<td align="right">0.0078669</td>
<td align="right">0.0018710</td>
</tr>
<tr class="odd">
<td align="left">chr1:1817875:1825397:clu_1001_-</td>
<td align="right">-0.0848513</td>
<td align="right">0.5066623</td>
<td align="right">0.5088598</td>
<td align="right">0.0021974</td>
</tr>
<tr class="even">
<td align="left">chr1:1825499:1839190:clu_1002_-</td>
<td align="right">0.0350052</td>
<td align="right">0.5886054</td>
<td align="right">0.6221619</td>
<td align="right">0.0335565</td>
</tr>
</tbody>
</table>
<p>Get a sense of number of intron clusters tested, number significant, etc</p>
<pre class="r"><code>#how many clusters tested (&quot;Success&quot;)
table(sig$status)</code></pre>
<pre><code>
&lt;=1 sample with coverage&gt;min_coverage              Not enough valid samples 
                                  212                                  1278 
                              Success 
                                 3619 </code></pre>
<pre class="r"><code>#histogram of Pvalues
hist(sig$p)</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#how many significant (p.adjust &lt; 0.1)
table(sig$p.adjust&lt;0.1)</code></pre>
<pre><code>
FALSE  TRUE 
 2860   759 </code></pre>
<pre class="r"><code># merge significance and cluster tables
leafcutter.merged &lt;- effect_sizes %&gt;%
  mutate(cluster=gsub(&quot;(.+?:).+?:.+?:(clu.+?)&quot;, &quot;\\1\\2&quot;, intron, perl=T)) %&gt;%
  mutate(junc_id=gsub(&quot;(.+?:.+?:.+?):clu.+&quot;, &quot;\\1&quot;, intron, perl=T)) %&gt;%
  left_join(sig, by=&quot;cluster&quot;)

# volcano plot of cluster-pvalues and largest within-cluster-delta-psi
leafcutter.merged %&gt;%
  mutate(abs.deltapsi = abs(deltapsi)) %&gt;%
  group_by(cluster) %&gt;% 
  top_n(n=1, abs.deltapsi) %&gt;%
  mutate(sig=p.adjust&lt;0.05) %&gt;%
  ggplot(aes(x=deltapsi, y=-log10(p), color=sig)) +
    geom_point(alpha=0.05) +
    theme_bw()</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/0d7521b0051dd8fd31ed5857d788357da9fdd302/docs/figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-2.png" target="_blank">0d7521b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-3-2.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>let’s do a gsea to see what genes have differentially spliced clusters of introns. For this, we need an ordered gene list, with each gene appearing once. Since the biological question related to this is what are the genes or gene sets that are getting disrupted to cause the organismal phenotype, i think it is important to consider not just significance (P-value) but also effect size, perhaps as measured by delta-psi. Since significane and effect size can be quite different in a splicing analysis (see volcano plot above) because of low read counts on some junctions, let’s first filter out all non-significant splicing changes (to filter out the noisy effect size estimates) Then let’s make an ordered list based on the maximum delta-PSI for each gene.</p>
<pre class="r"><code>#let&#39;s use a fairly permissive significance threshold to be more inclusive. how about FDR&gt;0.3
hist(leafcutter.merged$p.adjust)</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/c1d660db6d38073970096cd71db8fa190693c0d4/docs/figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-4-1.png" target="_blank">c1d660d</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#filter and get max delta-psi for each gene. Keep in mind we need to filter out non-valid genes for the clusters or introns that have more than one gene listed in the genes column
OrderedGeneList &lt;- leafcutter.merged %&gt;%
  filter(p.adjust&lt;0.3) %&gt;%
  mutate(abs.deltapsi = abs(deltapsi)) %&gt;%
  group_by(genes) %&gt;% 
  top_n(n=1, abs.deltapsi) %&gt;%
  filter(!grepl(&quot;,&quot;,genes)) %&gt;%
  arrange(desc(abs.deltapsi)) %&gt;%
  dplyr::select(genes, abs.deltapsi) %&gt;%
  deframe()

head(OrderedGeneList)</code></pre>
<pre><code>    PQBP1    FIGNL1    KANSL1    KANSL1    RNF220     PARP2 
0.9807278 0.6522834 0.5597076 0.5597076 0.5443519 0.5362952 </code></pre>
<pre class="r"><code>gsea.results &lt;- gseGO(OrderedGeneList, ont=&quot;ALL&quot;, OrgDb=org.Hs.eg.db, keyType=&#39;SYMBOL&#39;, nPerm=100000)

gsea.results.df &lt;- as.data.frame(gsea.results)

dim(gsea.results.df)</code></pre>
<pre><code>[1] 0 9</code></pre>
<p>No significant GSEA results at the default P.adjust&lt;0.5 threshold.</p>
<p>Ok, besides identifying the genes (or kinds of genes) through which the disease phenotype is potentially mediated, maybe the other questions I should be asking is what kinds of intron features distinguish differentially spliced introns, which may be more enlightening about general splicing mechanisms in these DHX38 mutants.</p>
<p>I have scored the 5’ss and 3’ss motifs for each intron tested using a position specific score matrix based on annotated introns. I additionally used <a href="http://regulatorygenomics.upf.edu/Software/SVM_BP/">svm-bpfinder</a> to score the best computationally predicted branch for each intron. I also annotated whether each intron is annotated, alt3’ss, alt5’ss, new pairing of splice sites, or new intron. All of that work is hidden in the <code>code</code> section and output into a file in <code>output</code>.</p>
<p>Let’s see if any of those features correlate with differentially expressed introns. If there is the DHX38 mutations preferentially activates less consensus splice sites for example, I expect the directionality to be consistent. Therefore, I will correlate those scores with either the logFC for each intron, for significant introns.</p>
<pre class="r"><code>#read intron features
IntronFeatures &lt;- read.table(&quot;../output/IntronFeatures.txt.gz&quot;, sep=&#39;\t&#39;, header=T)

head(IntronFeatures) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Intron</th>
<th align="left">strand</th>
<th align="left">type</th>
<th align="left">donor_seq</th>
<th align="right">DonorScore</th>
<th align="left">acceptor_seq</th>
<th align="right">AcceptorScore</th>
<th align="right">eqez</th>
<th align="right">ss_dist</th>
<th align="left">bp_seq</th>
<th align="right">bp_scr</th>
<th align="right">y_cont</th>
<th align="right">ppt_off</th>
<th align="right">ppt_len</th>
<th align="right">ppt_scr</th>
<th align="right">svm_scr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1_11671_12009_+</td>
<td align="left">+</td>
<td align="left">NewIntron</td>
<td align="left">CAGGTGAAG</td>
<td align="right">7.509</td>
<td align="left">GCTGTGGTCTTCATCTGCAGGTG</td>
<td align="right">9.120</td>
<td align="right">29</td>
<td align="right">22</td>
<td align="left">tcttgatgc</td>
<td align="right">0.7736716</td>
<td align="right">0.7058824</td>
<td align="right">9</td>
<td align="right">9</td>
<td align="right">19</td>
<td align="right">0.5400024</td>
</tr>
<tr class="even">
<td align="left">1_11844_12009_+</td>
<td align="left">+</td>
<td align="left">NewIntron</td>
<td align="left">CGGGTATCA</td>
<td align="right">2.853</td>
<td align="left">GCTGTGGTCTTCATCTGCAGGTG</td>
<td align="right">9.120</td>
<td align="right">29</td>
<td align="right">22</td>
<td align="left">tcttgatgc</td>
<td align="right">0.7736716</td>
<td align="right">0.7058824</td>
<td align="right">9</td>
<td align="right">9</td>
<td align="right">19</td>
<td align="right">0.5400024</td>
</tr>
<tr class="odd">
<td align="left">1_12721_13220_+</td>
<td align="left">+</td>
<td align="left">AnnotatedSpliceSite</td>
<td align="left">GAGGTGAGA</td>
<td align="right">9.498</td>
<td align="left">TAGCCTCTGTTCCCACGAAGGCA</td>
<td align="right">4.753</td>
<td align="right">17</td>
<td align="right">19</td>
<td align="left">gccttagcc</td>
<td align="right">-0.9155434</td>
<td align="right">0.7857143</td>
<td align="right">2</td>
<td align="right">13</td>
<td align="right">22</td>
<td align="right">0.3754140</td>
</tr>
<tr class="even">
<td align="left">1_14829_14929_-</td>
<td align="left">-</td>
<td align="left">NewIntron</td>
<td align="left">CTTGTAATT</td>
<td align="right">3.934</td>
<td align="left">ACTCAGCCTTTTCCCTCCAGGCA</td>
<td align="right">11.310</td>
<td align="right">14</td>
<td align="right">20</td>
<td align="left">ccctcactc</td>
<td align="right">2.5121196</td>
<td align="right">0.8666667</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">28</td>
<td align="right">1.8628447</td>
</tr>
<tr class="odd">
<td align="left">1_14829_15020_-</td>
<td align="left">-</td>
<td align="left">NewIntron</td>
<td align="left">AAGGTGGGC</td>
<td align="right">7.731</td>
<td align="left">ACTCAGCCTTTTCCCTCCAGGCA</td>
<td align="right">11.310</td>
<td align="right">14</td>
<td align="right">20</td>
<td align="left">ccctcactc</td>
<td align="right">2.5121196</td>
<td align="right">0.8666667</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">28</td>
<td align="right">1.8628447</td>
</tr>
<tr class="even">
<td align="left">1_14829_14969_-</td>
<td align="left">-</td>
<td align="left">NewIntron</td>
<td align="left">AGGGTAGGA</td>
<td align="right">6.363</td>
<td align="left">ACTCAGCCTTTTCCCTCCAGGCA</td>
<td align="right">11.310</td>
<td align="right">14</td>
<td align="right">20</td>
<td align="left">ccctcactc</td>
<td align="right">2.5121196</td>
<td align="right">0.8666667</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">28</td>
<td align="right">1.8628447</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Check that the number of types of unannotated splicing is similar in both strands. Sometimes bugs related to coordinates produce wonky results
IntronFeatures %&gt;%
  filter(strand==&quot;-&quot;) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
              23770               18569               14989              105919 
          NewIntron 
              22055 </code></pre>
<pre class="r"><code>IntronFeatures %&gt;%
  filter(strand==&quot;+&quot;) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
              24451               18695               15300              108186 
          NewIntron 
              23013 </code></pre>
<pre class="r"><code># merge the intron features with the leafcutter results and filter for the significantly differentially spliced introns that have the greatest effect size within cluster
leafcutter.merged.w.features &lt;- leafcutter.merged %&gt;%
  separate(junc_id, c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;), sep = &quot;:&quot;) %&gt;%
  mutate(newChrom=sub(&quot;chr&quot;, &quot;&quot;, chrom),
         newStop=as.numeric(stop)-1,
         strand=str_sub(intron, start=-1)) %&gt;%
  mutate(IntronID=str_c(newChrom, start, newStop, strand, sep=&quot;_&quot;)) %&gt;%
  dplyr::select(-c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;newStop&quot;, &quot;newChrom&quot;)) %&gt;%
  mutate(abs.effectsize = abs(logef)) %&gt;%
  group_by(cluster) %&gt;% 
  top_n(n=1, abs.effectsize) %&gt;%
  ungroup() %&gt;%
  filter(p.adjust&lt;0.2) %&gt;%
  left_join(IntronFeatures, by=c(&quot;IntronID&quot;=&quot;Intron&quot;))

# Are specific types of unannotated splicing more prevalent?
leafcutter.merged.w.features %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
                 68                  34                  55                1406 
          NewIntron 
                 10 </code></pre>
<pre class="r"><code>colnames(leafcutter.merged.w.features)</code></pre>
<pre><code> [1] &quot;intron&quot;         &quot;logef&quot;          &quot;H&quot;              &quot;A&quot;             
 [5] &quot;deltapsi&quot;       &quot;cluster&quot;        &quot;status&quot;         &quot;loglr&quot;         
 [9] &quot;df&quot;             &quot;p&quot;              &quot;p.adjust&quot;       &quot;genes&quot;         
[13] &quot;strand.x&quot;       &quot;IntronID&quot;       &quot;abs.effectsize&quot; &quot;strand.y&quot;      
[17] &quot;type&quot;           &quot;donor_seq&quot;      &quot;DonorScore&quot;     &quot;acceptor_seq&quot;  
[21] &quot;AcceptorScore&quot;  &quot;eqez&quot;           &quot;ss_dist&quot;        &quot;bp_seq&quot;        
[25] &quot;bp_scr&quot;         &quot;y_cont&quot;         &quot;ppt_off&quot;        &quot;ppt_len&quot;       
[29] &quot;ppt_scr&quot;        &quot;svm_scr&quot;       </code></pre>
<pre class="r"><code># Set up features and effect sizes into a cleaner data frame
IntronFeaturesAndEffectSizes &lt;- leafcutter.merged.w.features %&gt;%
  dplyr::select(IntronID, logef, DonorScore, AcceptorScore, ss_dist, bp_scr, ppt_len, ppt_scr, svm_scr) %&gt;%
  column_to_rownames(&quot;IntronID&quot;)

# See if there is any correlations
CorrelationsWithFeatures &lt;- corr.test(x=IntronFeaturesAndEffectSizes[,1],
                         y=IntronFeaturesAndEffectSizes[,-1],
                         method=&quot;spearman&quot;,
                         adjust=&quot;none&quot;)

# table of spearmans r, pvalue, and n
data.frame(r= t(CorrelationsWithFeatures$r), p=t(CorrelationsWithFeatures$p), n=t(CorrelationsWithFeatures$n)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="center">r</th>
<th align="center">p</th>
<th align="center">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DonorScore</td>
<td align="center">-0.0254915</td>
<td align="center">0.3123134</td>
<td align="center">1573</td>
</tr>
<tr class="even">
<td>AcceptorScore</td>
<td align="center">0.0164397</td>
<td align="center">0.5146981</td>
<td align="center">1573</td>
</tr>
<tr class="odd">
<td>ss_dist</td>
<td align="center">0.0098841</td>
<td align="center">0.7060034</td>
<td align="center">1459</td>
</tr>
<tr class="even">
<td>bp_scr</td>
<td align="center">-0.0123674</td>
<td align="center">0.6369214</td>
<td align="center">1459</td>
</tr>
<tr class="odd">
<td>ppt_len</td>
<td align="center">0.0214754</td>
<td align="center">0.4123965</td>
<td align="center">1459</td>
</tr>
<tr class="even">
<td>ppt_scr</td>
<td align="center">0.0160727</td>
<td align="center">0.5395862</td>
<td align="center">1459</td>
</tr>
<tr class="odd">
<td>svm_scr</td>
<td align="center">-0.0138876</td>
<td align="center">0.5960885</td>
<td align="center">1459</td>
</tr>
</tbody>
</table>
<p>None of the features tested seem to correlate with the effect size of largest-effect introns in significant clusters.</p>
<p>Check if the significant positive effect-size splicing changes have more cryptic 5’ss, or more cryptic 3’ss than negative effect-size splicing changes.</p>
<p>When doing this analysis, some people think in terms of deltapsi, some people might think in terms of a different metric of effect size. Let’s look at both.</p>
<pre class="r"><code># leafcutter&#39;s logef effect size, versus deltapsi
qplot(leafcutter.merged.w.features$logef, leafcutter.merged.w.features$deltapsi) + theme_bw()</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/0d7521b0051dd8fd31ed5857d788357da9fdd302/docs/figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-6-1.png" target="_blank">0d7521b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#significant positive deltapsi introns
leafcutter.merged.w.features %&gt;%
  filter(deltapsi &gt; 0) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
                 28                  12                  25                 709 
          NewIntron 
                  6 </code></pre>
<pre class="r"><code>#significant negative deltapsi introns
leafcutter.merged.w.features %&gt;%
  filter(deltapsi &lt; 0) %&gt;%
  pull(type) %&gt;% table()</code></pre>
<pre><code>.
             Alt3ss              Alt5ss   AltPairingOfSites AnnotatedSpliceSite 
                 40                  22                  30                 697 
          NewIntron 
                  4 </code></pre>
<pre class="r"><code>leafcutter.merged.w.features %&gt;%
  ggplot(aes(x=deltapsi, color=type)) +
    stat_ecdf(geom = &quot;step&quot;) +
    ylab(&quot;Cumulative frequency&quot;) +
    theme_bw()</code></pre>
<p><img src="figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/rna-seq-dhx38/blob/0d7521b0051dd8fd31ed5857d788357da9fdd302/docs/figure/20191203_DifferentialSplicing.Rmd/unnamed-chunk-6-2.png" target="_blank">0d7521b</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2019-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Maybe 5’ss stick out? Let’s test it more explicitly.</p>
<p>The null hypothesis here is that any given type of splicing will not significantly deviate from median of 0 effect size (similar number of positive effects as negative effects). Wilcox.test to test with null as mu=0 should work.</p>
<pre class="r"><code>## are alt 5&#39;ss upregulated?
wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;Alt5ss&quot;) %&gt;% pull(deltapsi)
    )</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;Alt5ss&quot;) %&gt;%     pull(deltapsi)
V = 150.5, p-value = 0.01226
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;Alt5ss&quot;) %&gt;% pull(logef)
    )</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;Alt5ss&quot;) %&gt;%     pull(logef)
V = 166.5, p-value = 0.02567
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code>## are alt 3&#39;ss upregulated?
wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;Alt3ss&quot;) %&gt;% pull(deltapsi)
    )</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;Alt3ss&quot;) %&gt;%     pull(deltapsi)
V = 1001, p-value = 0.2947
alternative hypothesis: true location is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(
  leafcutter.merged.w.features %&gt;%
  filter(type==&quot;Alt3ss&quot;) %&gt;% pull(logef)
    )</code></pre>
<pre><code>
    Wilcoxon signed rank test with continuity correction

data:  leafcutter.merged.w.features %&gt;% filter(type == &quot;Alt3ss&quot;) %&gt;%     pull(logef)
V = 1058, p-value = 0.4842
alternative hypothesis: true location is not equal to 0</code></pre>
<p>Hmm. maybe there is something here with alt 5’ss. Considering all the exploration I’ve been doing, and without multiple hypothesis correction in many cases this might or might not be real.</p>
</div>
<div id="todo" class="section level2">
<h2>TODO</h2>
<ul>
<li>should look at the alt 5’ss and see if there is any distance constraint with the annotated 5’ss, similar to the SF3B1 mutants that have a very specific distance constraint to the usual splice sites that get activated</li>
</ul>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] corrplot_0.84          psych_1.8.12           enrichplot_1.4.0      
 [4] org.Hs.eg.db_3.8.2     AnnotationDbi_1.46.1   IRanges_2.18.3        
 [7] S4Vectors_0.22.1       Biobase_2.44.0         BiocGenerics_0.30.0   
[10] clusterProfiler_3.12.0 biomaRt_2.40.5         edgeR_3.26.8          
[13] limma_3.40.6           knitr_1.26             forcats_0.4.0         
[16] stringr_1.4.0          dplyr_0.8.3            purrr_0.3.3           
[19] readr_1.3.1            tidyr_1.0.0            tibble_2.1.3          
[22] ggplot2_3.2.1          tidyverse_1.2.1       

loaded via a namespace (and not attached):
  [1] fgsea_1.10.1        colorspace_1.4-1    ellipsis_0.3.0     
  [4] ggridges_0.5.1      rprojroot_1.3-2     qvalue_2.16.0      
  [7] fs_1.3.1            rstudioapi_0.10     farver_1.1.0       
 [10] urltools_1.7.3      graphlayouts_0.5.0  ggrepel_0.8.1      
 [13] bit64_0.9-7         lubridate_1.7.4     xml2_1.2.2         
 [16] splines_3.6.1       mnormt_1.5-5        GOSemSim_2.10.0    
 [19] polyclip_1.10-0     zeallot_0.1.0       jsonlite_1.6       
 [22] workflowr_1.5.0     broom_0.5.2         GO.db_3.8.2        
 [25] ggforce_0.3.1       BiocManager_1.30.9  compiler_3.6.1     
 [28] httr_1.4.1          rvcheck_0.1.6       backports_1.1.5    
 [31] assertthat_0.2.1    Matrix_1.2-17       lazyeval_0.2.2     
 [34] cli_1.1.0           later_1.0.0         tweenr_1.0.1       
 [37] htmltools_0.4.0     prettyunits_1.0.2   tools_3.6.1        
 [40] igraph_1.2.4.1      gtable_0.3.0        glue_1.3.1         
 [43] reshape2_1.4.3      DO.db_2.9           fastmatch_1.1-0    
 [46] Rcpp_1.0.2          cellranger_1.1.0    vctrs_0.2.0        
 [49] nlme_3.1-142        ggraph_2.0.0        xfun_0.11          
 [52] rvest_0.3.5         lifecycle_0.1.0     XML_3.98-1.20      
 [55] DOSE_3.10.2         europepmc_0.3       MASS_7.3-51.4      
 [58] scales_1.0.0        tidygraph_1.1.2     hms_0.5.2          
 [61] promises_1.1.0      RColorBrewer_1.1-2  yaml_2.2.0         
 [64] memoise_1.1.0       gridExtra_2.3       UpSetR_1.4.0       
 [67] triebeard_0.3.0     stringi_1.4.3       RSQLite_2.1.2      
 [70] highr_0.8           BiocParallel_1.18.1 rlang_0.4.1        
 [73] pkgconfig_2.0.3     bitops_1.0-6        evaluate_0.14      
 [76] lattice_0.20-38     labeling_0.3        cowplot_1.0.0      
 [79] bit_1.1-14          tidyselect_0.2.5    plyr_1.8.4         
 [82] magrittr_1.5        R6_2.4.1            generics_0.0.2     
 [85] DBI_1.0.0           foreign_0.8-72      pillar_1.4.2       
 [88] haven_2.2.0         whisker_0.4         withr_2.1.2        
 [91] RCurl_1.95-4.12     modelr_0.1.5        crayon_1.3.4       
 [94] rmarkdown_1.17      viridis_0.5.1       progress_1.2.2     
 [97] locfit_1.5-9.1      grid_3.6.1          readxl_1.3.1       
[100] data.table_1.12.6   blob_1.2.0          git2r_0.26.1       
[103] digest_0.6.22       httpuv_1.5.2        gridGraphics_0.4-1 
[106] munsell_0.5.0       ggplotify_0.0.4     viridisLite_0.3.0  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
