<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2019-12-16" />

<title>Overlap with diff polyA length</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Overlap with diff polyA length</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">December 16, 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-12-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rna-seq-dhx38/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38treed602320dfafe088c010d645b5a782854b0cc40e4targetblankd602320a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/rna-seq-dhx38/tree/d602320dfafe088c010d645b5a782854b0cc40e4" target="_blank">d602320</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38treed602320dfafe088c010d645b5a782854b0cc40e4targetblankd602320a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/20191216_DifferentialExpression_Splicing.Rmd
    Untracked:  analysis/20191216_polyA_Intersect.Rmd
    Untracked:  data/Diff_tail_length_WT_vs_DHX.csv
    Untracked:  data/GO_splicing_ids.fromAmiGO_online_search.tab
    Untracked:  output/BioMartQuery.Go.Splicing.tab

Unstaged changes:
    Modified:   analysis/20191203_DiffExpression.Rmd
    Modified:   analysis/20191203_DifferentialSplicing.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>As I understand it (based on reading email correspondences and filenames), the collaborators also took PacBio-based measurements of polyA length between healthy and affected and performed a differential polyA length expression analysis. ## Analysis</p>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(knitr)</code></pre>
<p>Read in data from collaborator (some sort of differential analysis based on PacBio data)</p>
<pre class="r"><code>#Read PacBioResults
PacBio.results &lt;- read.csv(&quot;../data/Diff_tail_length_WT_vs_DHX.csv&quot;, )
head(PacBio.results) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">X</th>
<th align="left">gene</th>
<th align="right">cntA</th>
<th align="right">cntB</th>
<th align="right">med_pa_len_A</th>
<th align="right">med_pa_len_B</th>
<th align="right">iso_pa_pval</th>
<th align="left">iso_pa_sig</th>
<th align="right">iso_pa_diff</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">AAAS</td>
<td align="right">24</td>
<td align="right">24</td>
<td align="right">129.5</td>
<td align="right">141.5</td>
<td align="right">0.4491726</td>
<td align="left">False</td>
<td align="right">12.0</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="left">AAGAB</td>
<td align="right">8</td>
<td align="right">12</td>
<td align="right">80.0</td>
<td align="right">104.5</td>
<td align="right">0.2935308</td>
<td align="left">False</td>
<td align="right">24.5</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="left">AAMP</td>
<td align="right">37</td>
<td align="right">53</td>
<td align="right">79.0</td>
<td align="right">110.0</td>
<td align="right">0.2071186</td>
<td align="left">False</td>
<td align="right">31.0</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="left">AARS</td>
<td align="right">13</td>
<td align="right">8</td>
<td align="right">130.0</td>
<td align="right">137.0</td>
<td align="right">0.4763985</td>
<td align="left">False</td>
<td align="right">7.0</td>
</tr>
<tr class="odd">
<td align="right">8</td>
<td align="left">AASDHPPT</td>
<td align="right">17</td>
<td align="right">15</td>
<td align="right">133.0</td>
<td align="right">87.0</td>
<td align="right">0.1009361</td>
<td align="left">False</td>
<td align="right">46.0</td>
</tr>
<tr class="even">
<td align="right">9</td>
<td align="left">AATF</td>
<td align="right">54</td>
<td align="right">61</td>
<td align="right">68.0</td>
<td align="right">82.0</td>
<td align="right">0.4151206</td>
<td align="left">False</td>
<td align="right">14.0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Read standard RNAseq results
DiffExp &lt;- read.table(&quot;../output/GeneExpression.DE.tab&quot;, stringsAsFactors = F, header=T, sep=&#39;\t&#39;)

head(DiffExp) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
<th align="left">external_gene_name</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000279457</td>
<td align="right">-0.0019184</td>
<td align="right">2.294509</td>
<td align="right">1.0000000</td>
<td align="right">1.0000000</td>
<td align="left">WASH9P</td>
<td align="left">WAS protein family homolog 9, pseudogene [Source:HGNC Symbol;Acc:HGNC:53981]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000248527</td>
<td align="right">-1.4384558</td>
<td align="right">3.327765</td>
<td align="right">0.0005842</td>
<td align="right">0.0028015</td>
<td align="left">MTATP6P1</td>
<td align="left">MT-ATP6 pseudogene 1 [Source:HGNC Symbol;Acc:HGNC:44575]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000228794</td>
<td align="right">0.0824910</td>
<td align="right">3.223395</td>
<td align="right">0.6820905</td>
<td align="right">0.7909175</td>
<td align="left">LINC01128</td>
<td align="left">long intergenic non-protein coding RNA 1128 [Source:HGNC Symbol;Acc:HGNC:49377]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000230699</td>
<td align="right">0.5618857</td>
<td align="right">2.490896</td>
<td align="right">0.0208296</td>
<td align="right">0.0580221</td>
<td align="left">AL645608.2</td>
<td align="left">novel transcript</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000188976</td>
<td align="right">-0.1738289</td>
<td align="right">6.991208</td>
<td align="right">0.2377004</td>
<td align="right">0.3809051</td>
<td align="left">NOC2L</td>
<td align="left">NOC2 like nucleolar associated transcriptional repressor [Source:HGNC Symbol;Acc:HGNC:24517]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000187961</td>
<td align="right">-0.0084449</td>
<td align="right">4.092404</td>
<td align="right">0.9652432</td>
<td align="right">0.9806255</td>
<td align="left">KLHL17</td>
<td align="left">kelch like family member 17 [Source:HGNC Symbol;Acc:HGNC:24023]</td>
</tr>
</tbody>
</table>
<pre class="r"><code>Merged &lt;- inner_join(PacBio.results, DiffExp, by=c(&quot;gene&quot;=&quot;external_gene_name&quot;))</code></pre>
<p>Let’s visualize the data a bit…</p>
<p>First, a correlation of polyA length between conditions</p>
<pre class="r"><code>Merged %&gt;%
ggplot(aes(x=med_pa_len_A, y=med_pa_len_B, color=iso_pa_sig)) +
  geom_point(alpha=0.3) +
  theme_bw()</code></pre>
<p><img src="figure/20191216_polyA_Intersect.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From browsing the PacBio table, I believe iso_pa diff is just the absolute value of the difference between med_pa_len_[A/B]. let’s check that</p>
<pre class="r"><code>Merged %&gt;%
ggplot(aes(x=med_pa_len_A-med_pa_len_B, y=iso_pa_diff,color=iso_pa_sig)) +
  geom_point(alpha=0.3) +
  theme_bw()</code></pre>
<p><img src="figure/20191216_polyA_Intersect.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok. I think the cnt columns refer to read counts (expression). Let’s try to recreate the MA plot I think I saw in the email correspondence</p>
<pre class="r"><code>ggplot(PacBio.results, aes(x=log2((cntA+cntB)/2), y=med_pa_len_A-med_pa_len_B, color=iso_pa_sig)) +
  geom_point() +
  theme_bw()</code></pre>
<p><img src="figure/20191216_polyA_Intersect.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok yeah, that looks a lot like the plot I saw in the email correspondence, so I think I am understanding the PacBio results correctly.</p>
<p>To make sure, let’s check that the cntA and cntB (expression) correlates with expression as measured by RNA-seq (Technically these aren’t perfectly comparable since the RNA-seq expression is measured in CountsPerMillion, rather than RPKM, so there is no length normalization). Nonetheless, I expect to see some correlation.</p>
<pre class="r"><code>ggplot(Merged, aes(x=log2((cntA+cntB)/2), y=logCPM)) +
  geom_point() +
  theme_bw()</code></pre>
<p><img src="figure/20191216_polyA_Intersect.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cor.test(log2((Merged$cntA+Merged$cntB)/2), Merged$logCPM, method=&#39;spearman&#39;)</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  log2((Merged$cntA + Merged$cntB)/2) and Merged$logCPM
S = 739931301, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.2513013 </code></pre>
<p>Now, is there any correlation between expression polyA length and expression? (The natural hypothesis is that the genes with longer tails will be stabilized, and therefore upregulated)</p>
<pre class="r"><code>Merged %&gt;%
ggplot(aes(x=med_pa_len_A-med_pa_len_B, y=logFC,color=iso_pa_sig)) +
  geom_point(alpha=0.3) +
  theme_bw() +
  xlab(&quot;Difference in polyA length&quot;) +
  ylab(&quot;logFC expression&quot;)</code></pre>
<p><img src="figure/20191216_polyA_Intersect.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cor.test(Merged$med_pa_len_A-Merged$med_pa_len_B, Merged$logFC, method=&#39;spearman&#39;)</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  Merged$med_pa_len_A - Merged$med_pa_len_B and Merged$logFC
S = 974728825, p-value = 0.5596
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
0.01372172 </code></pre>
<p>No obvious difference and no significant correlation. Let’s check if there is any significant overlap between the sets of significant gene by fisher test of the 2x2 contigency table of significant vs not significant gene count for each method.</p>
<pre class="r"><code>Merged$ExpressionSig &lt;- Merged$FDR&lt;0.1

#how many significant by each method?
table(Merged$ExpressionSig)</code></pre>
<pre><code>
FALSE  TRUE 
 1175   635 </code></pre>
<pre class="r"><code>table(Merged$iso_pa_sig)</code></pre>
<pre><code>
False  True 
 1650   160 </code></pre>
<pre class="r"><code>Merged %&gt;%
  dplyr::select(ExpressionSig, iso_pa_sig) %&gt;%
  group_by(ExpressionSig, iso_pa_sig) %&gt;% summarise(n = n()) %&gt;%
  spread(key = ExpressionSig, value = n) %&gt;%
  column_to_rownames(&quot;iso_pa_sig&quot;) %&gt;%
  fisher.test()</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  .
p-value = 0.6645
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.7636369 1.5424306
sample estimates:
odds ratio 
  1.089158 </code></pre>
<p>Ok the overlap is not significant. As far as we can measure, the significance of these assays seem to be uncorrelated.</p>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] knitr_1.26      forcats_0.4.0   stringr_1.4.0   dplyr_0.8.3    
 [5] purrr_0.3.3     readr_1.3.1     tidyr_1.0.0     tibble_2.1.3   
 [9] ggplot2_3.2.1   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 xfun_0.11        haven_2.2.0      lattice_0.20-38 
 [5] colorspace_1.4-1 vctrs_0.2.0      generics_0.0.2   htmltools_0.4.0 
 [9] yaml_2.2.0       rlang_0.4.1      later_1.0.0      pillar_1.4.2    
[13] withr_2.1.2      glue_1.3.1       DBI_1.0.0        dbplyr_1.4.2    
[17] modelr_0.1.5     readxl_1.3.1     lifecycle_0.1.0  munsell_0.5.0   
[21] gtable_0.3.0     workflowr_1.5.0  cellranger_1.1.0 rvest_0.3.5     
[25] evaluate_0.14    labeling_0.3     httpuv_1.5.2     fansi_0.4.0     
[29] highr_0.8        broom_0.5.2      Rcpp_1.0.2       promises_1.1.0  
[33] backports_1.1.5  scales_1.1.0     jsonlite_1.6     farver_2.0.1    
[37] fs_1.3.1         hms_0.5.2        digest_0.6.23    stringi_1.4.3   
[41] grid_3.6.1       rprojroot_1.3-2  cli_2.0.0        tools_3.6.1     
[45] magrittr_1.5     lazyeval_0.2.2   crayon_1.3.4     pkgconfig_2.0.3 
[49] zeallot_0.1.0    xml2_1.2.2       reprex_0.3.0     lubridate_1.7.4 
[53] rstudioapi_0.10  assertthat_0.2.1 rmarkdown_1.18   httr_1.4.1      
[57] R6_2.4.1         nlme_3.1-143     git2r_0.26.1     compiler_3.6.1  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
