<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Benjamin Fair" />

<meta name="date" content="2020-01-14" />

<title>Untitled</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Untitled</h1>
<h4 class="author">Benjamin Fair</h4>
<h4 class="date">January 14, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-01-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rna-seq-dhx38/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191126code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191126)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191126code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191126)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree2ce85b8d6032575c1844103c3da6463192639abetargetblank2ce85b8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/rna-seq-dhx38/tree/2ce85b8d6032575c1844103c3da6463192639abe" target="_blank">2ce85b8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunrnaseqdhx38tree2ce85b8d6032575c1844103c3da6463192639abetargetblank2ce85b8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/20200110_DiffExpression_MoreReplicates.Rmd
    Untracked:  output/GeneExpression.DE.WithMoreReplicates.tab
    Untracked:  output/GeneExpression.gsea.WithMoreReplicates.tab

Unstaged changes:
    Modified:   analysis/20190105_DifferentialSplicingMoreReplicates.Rmd
    Modified:   analysis/20191203_DiffExpression.Rmd
    Modified:   output/GeneExpression.DE.tab
    Modified:   output/GeneExpression.gsea.tab

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>I previously did a differential splicing analysis with 3 replicates of affected and 3 replicates of unrelated healthy. Now we have more healthy datasets of other unrelated individuals for more robust analysis to identify effects that might be unqiue to the DHX38 mutations.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Raw RNA-seq data has been aligned and collapsed into gene count table (rows of genes, columns of samples, each cell is exonic read counts) using code in <code>code</code> section of this repo, creating a count matrix in <code>output</code>. Here I will process that data with edgeR approach that I have used previously. The combined samples are simply biological replicates combined. Since there are multiple levels of replication, I think a reasonable way to do this analysis is to just combine the bioligical replicates and do a 1 affected vs 3 healthy analysis.</p>
<p>First, load necessary libraries</p>
<pre class="r"><code>library(tidyverse)
library(knitr)
library(edgeR)
library(corrplot)
library(gplots)
library(biomaRt)
library(ggrepel)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(enrichplot)</code></pre>
<p>And read in the count table</p>
<pre class="r"><code>CountTable &lt;- read.table(&#39;../output/GeneExpressionCountTable.subread.txt.gz&#39;, header=T, check.names=FALSE, skip=1)


CountTable.matrix &lt;- CountTable %&gt;%
  dplyr::select(-c(&quot;Chr&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;Strand&quot;, &quot;Length&quot;)) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;% as.matrix()

#Preview table
kable(CountTable.matrix[1:10,])</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Healthy.JG_3</th>
<th align="right">Healthy.BIH4_1</th>
<th align="right">Healthy_1</th>
<th align="right">Healthy_3</th>
<th align="right">Combined.BIH4</th>
<th align="right">Affected_2</th>
<th align="right">Affected_1</th>
<th align="right">Combined.XM001</th>
<th align="right">Healthy_2</th>
<th align="right">Healthy.JG_2</th>
<th align="right">Healthy.BIH4_2</th>
<th align="right">Combined.JG</th>
<th align="right">Combined.DHX</th>
<th align="right">Healthy.JG_1</th>
<th align="right">Affected_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000223972</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">4</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000227232</td>
<td align="right">30</td>
<td align="right">36</td>
<td align="right">30</td>
<td align="right">38</td>
<td align="right">83</td>
<td align="right">53</td>
<td align="right">43</td>
<td align="right">94</td>
<td align="right">26</td>
<td align="right">48</td>
<td align="right">47</td>
<td align="right">122</td>
<td align="right">130</td>
<td align="right">44</td>
<td align="right">34</td>
</tr>
<tr class="odd">
<td>ENSG00000278267</td>
<td align="right">16</td>
<td align="right">12</td>
<td align="right">14</td>
<td align="right">18</td>
<td align="right">25</td>
<td align="right">12</td>
<td align="right">13</td>
<td align="right">41</td>
<td align="right">9</td>
<td align="right">11</td>
<td align="right">13</td>
<td align="right">46</td>
<td align="right">36</td>
<td align="right">19</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td>ENSG00000243485</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000284332</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000237613</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000268020</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000240361</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000186092</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000238009</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<p>Plot total exonic gene counts per sample</p>
<pre class="r"><code>CountTable.matrix %&gt;% colSums() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(&quot;Sample&quot;) %&gt;%
  ggplot(aes(x=Sample, y=./1000000)) +
  geom_col() +
  ylab(&quot;Exonic read count (Millions)&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>As expected, the combined samples are the sum of the replicates.</p>
<p>Check correlations between samples</p>
<pre class="r"><code># CountTable to CountsPerMillion
cpm &lt;- cpm(CountTable.matrix, log=TRUE, prior.count=0.5)
kable(cpm[1:10,])</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Healthy.JG_3</th>
<th align="right">Healthy.BIH4_1</th>
<th align="right">Healthy_1</th>
<th align="right">Healthy_3</th>
<th align="right">Combined.BIH4</th>
<th align="right">Affected_2</th>
<th align="right">Affected_1</th>
<th align="right">Combined.XM001</th>
<th align="right">Healthy_2</th>
<th align="right">Healthy.JG_2</th>
<th align="right">Healthy.BIH4_2</th>
<th align="right">Combined.JG</th>
<th align="right">Combined.DHX</th>
<th align="right">Healthy.JG_1</th>
<th align="right">Affected_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000223972</td>
<td align="right">-2.8591246</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-4.0596960</td>
<td align="right">-4.6647839</td>
<td align="right">-4.4585544</td>
<td align="right">-3.6083115</td>
<td align="right">-3.6296760</td>
<td align="right">-2.6537340</td>
<td align="right">-3.0569103</td>
<td align="right">-4.1261628</td>
<td align="right">-2.6661101</td>
<td align="right">-4.3768956</td>
<td align="right">-2.2712367</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="even">
<td>ENSG00000227232</td>
<td align="right">0.8924541</td>
<td align="right">1.0623718</td>
<td align="right">0.7712198</td>
<td align="right">0.7833333</td>
<td align="right">1.0259928</td>
<td align="right">0.7020906</td>
<td align="right">0.5399739</td>
<td align="right">0.6410562</td>
<td align="right">0.3353232</td>
<td align="right">1.3433098</td>
<td align="right">0.9987444</td>
<td align="right">1.1295392</td>
<td align="right">0.5337226</td>
<td align="right">1.0894737</td>
<td align="right">0.2990411</td>
</tr>
<tr class="odd">
<td>ENSG00000278267</td>
<td align="right">-0.0042239</td>
<td align="right">-0.5019166</td>
<td align="right">-0.3138292</td>
<td align="right">-0.2807002</td>
<td align="right">-0.6806292</td>
<td align="right">-1.3959114</td>
<td align="right">-1.1517472</td>
<td align="right">-0.5380690</td>
<td align="right">-1.1630010</td>
<td align="right">-0.7536815</td>
<td align="right">-0.8272240</td>
<td align="right">-0.2613098</td>
<td align="right">-1.2800265</td>
<td align="right">-0.1086483</td>
<td align="right">-1.2925038</td>
</tr>
<tr class="even">
<td>ENSG00000243485</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-3.9708238</td>
<td align="right">-3.7198834</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-3.8864472</td>
<td align="right">-3.3297094</td>
<td align="right">-4.3021258</td>
<td align="right">-4.7520134</td>
<td align="right">-3.9864137</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="odd">
<td>ENSG00000284332</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="even">
<td>ENSG00000237613</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="odd">
<td>ENSG00000268020</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="even">
<td>ENSG00000240361</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="odd">
<td>ENSG00000186092</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
</tr>
<tr class="even">
<td>ENSG00000238009</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-6.0520744</td>
<td align="right">-4.7520134</td>
<td align="right">-6.0520744</td>
<td align="right">-3.5302110</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#plot cpm
median_log2_cpm &lt;- apply(cpm, 1, median)
hist(median_log2_cpm)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># check correlations between samples
CorMat &lt;- cor(cpm)
kable(CorMat)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Healthy.JG_3</th>
<th align="right">Healthy.BIH4_1</th>
<th align="right">Healthy_1</th>
<th align="right">Healthy_3</th>
<th align="right">Combined.BIH4</th>
<th align="right">Affected_2</th>
<th align="right">Affected_1</th>
<th align="right">Combined.XM001</th>
<th align="right">Healthy_2</th>
<th align="right">Healthy.JG_2</th>
<th align="right">Healthy.BIH4_2</th>
<th align="right">Combined.JG</th>
<th align="right">Combined.DHX</th>
<th align="right">Healthy.JG_1</th>
<th align="right">Affected_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Healthy.JG_3</td>
<td align="right">1.0000000</td>
<td align="right">0.9547294</td>
<td align="right">0.9551455</td>
<td align="right">0.9412465</td>
<td align="right">0.9617618</td>
<td align="right">0.9463351</td>
<td align="right">0.9450591</td>
<td align="right">0.9624578</td>
<td align="right">0.9448417</td>
<td align="right">0.9557623</td>
<td align="right">0.9527712</td>
<td align="right">0.9814294</td>
<td align="right">0.9586613</td>
<td align="right">0.9610586</td>
<td align="right">0.9534496</td>
</tr>
<tr class="even">
<td>Healthy.BIH4_1</td>
<td align="right">0.9547294</td>
<td align="right">1.0000000</td>
<td align="right">0.9539969</td>
<td align="right">0.9422676</td>
<td align="right">0.9872051</td>
<td align="right">0.9493463</td>
<td align="right">0.9476235</td>
<td align="right">0.9623758</td>
<td align="right">0.9456483</td>
<td align="right">0.9510469</td>
<td align="right">0.9588224</td>
<td align="right">0.9636047</td>
<td align="right">0.9614223</td>
<td align="right">0.9548696</td>
<td align="right">0.9560137</td>
</tr>
<tr class="odd">
<td>Healthy_1</td>
<td align="right">0.9551455</td>
<td align="right">0.9539969</td>
<td align="right">1.0000000</td>
<td align="right">0.9524657</td>
<td align="right">0.9596554</td>
<td align="right">0.9443687</td>
<td align="right">0.9438560</td>
<td align="right">0.9870490</td>
<td align="right">0.9562035</td>
<td align="right">0.9470934</td>
<td align="right">0.9506989</td>
<td align="right">0.9611764</td>
<td align="right">0.9555638</td>
<td align="right">0.9545264</td>
<td align="right">0.9495342</td>
</tr>
<tr class="even">
<td>Healthy_3</td>
<td align="right">0.9412465</td>
<td align="right">0.9422676</td>
<td align="right">0.9524657</td>
<td align="right">1.0000000</td>
<td align="right">0.9512527</td>
<td align="right">0.9554088</td>
<td align="right">0.9549189</td>
<td align="right">0.9807831</td>
<td align="right">0.9700262</td>
<td align="right">0.9375445</td>
<td align="right">0.9499229</td>
<td align="right">0.9465928</td>
<td align="right">0.9560525</td>
<td align="right">0.9414062</td>
<td align="right">0.9413015</td>
</tr>
<tr class="odd">
<td>Combined.BIH4</td>
<td align="right">0.9617618</td>
<td align="right">0.9872051</td>
<td align="right">0.9596554</td>
<td align="right">0.9512527</td>
<td align="right">1.0000000</td>
<td align="right">0.9634273</td>
<td align="right">0.9620600</td>
<td align="right">0.9702293</td>
<td align="right">0.9541399</td>
<td align="right">0.9626313</td>
<td align="right">0.9884341</td>
<td align="right">0.9744784</td>
<td align="right">0.9748043</td>
<td align="right">0.9652938</td>
<td align="right">0.9672163</td>
</tr>
<tr class="even">
<td>Affected_2</td>
<td align="right">0.9463351</td>
<td align="right">0.9493463</td>
<td align="right">0.9443687</td>
<td align="right">0.9554088</td>
<td align="right">0.9634273</td>
<td align="right">1.0000000</td>
<td align="right">0.9768898</td>
<td align="right">0.9620928</td>
<td align="right">0.9552610</td>
<td align="right">0.9531728</td>
<td align="right">0.9629860</td>
<td align="right">0.9602197</td>
<td align="right">0.9883522</td>
<td align="right">0.9537794</td>
<td align="right">0.9675438</td>
</tr>
<tr class="odd">
<td>Affected_1</td>
<td align="right">0.9450591</td>
<td align="right">0.9476235</td>
<td align="right">0.9438560</td>
<td align="right">0.9549189</td>
<td align="right">0.9620600</td>
<td align="right">0.9768898</td>
<td align="right">1.0000000</td>
<td align="right">0.9613420</td>
<td align="right">0.9544294</td>
<td align="right">0.9519131</td>
<td align="right">0.9620527</td>
<td align="right">0.9589687</td>
<td align="right">0.9871373</td>
<td align="right">0.9528822</td>
<td align="right">0.9664420</td>
</tr>
<tr class="even">
<td>Combined.XM001</td>
<td align="right">0.9624578</td>
<td align="right">0.9623758</td>
<td align="right">0.9870490</td>
<td align="right">0.9807831</td>
<td align="right">0.9702293</td>
<td align="right">0.9620928</td>
<td align="right">0.9613420</td>
<td align="right">1.0000000</td>
<td align="right">0.9835078</td>
<td align="right">0.9565234</td>
<td align="right">0.9637110</td>
<td align="right">0.9694140</td>
<td align="right">0.9697808</td>
<td align="right">0.9624151</td>
<td align="right">0.9597859</td>
</tr>
<tr class="odd">
<td>Healthy_2</td>
<td align="right">0.9448417</td>
<td align="right">0.9456483</td>
<td align="right">0.9562035</td>
<td align="right">0.9700262</td>
<td align="right">0.9541399</td>
<td align="right">0.9552610</td>
<td align="right">0.9544294</td>
<td align="right">0.9835078</td>
<td align="right">1.0000000</td>
<td align="right">0.9408971</td>
<td align="right">0.9513490</td>
<td align="right">0.9504304</td>
<td align="right">0.9575416</td>
<td align="right">0.9450088</td>
<td align="right">0.9443274</td>
</tr>
<tr class="even">
<td>Healthy.JG_2</td>
<td align="right">0.9557623</td>
<td align="right">0.9510469</td>
<td align="right">0.9470934</td>
<td align="right">0.9375445</td>
<td align="right">0.9626313</td>
<td align="right">0.9531728</td>
<td align="right">0.9519131</td>
<td align="right">0.9565234</td>
<td align="right">0.9408971</td>
<td align="right">1.0000000</td>
<td align="right">0.9568884</td>
<td align="right">0.9838795</td>
<td align="right">0.9650818</td>
<td align="right">0.9630776</td>
<td align="right">0.9590722</td>
</tr>
<tr class="odd">
<td>Healthy.BIH4_2</td>
<td align="right">0.9527712</td>
<td align="right">0.9588224</td>
<td align="right">0.9506989</td>
<td align="right">0.9499229</td>
<td align="right">0.9884341</td>
<td align="right">0.9629860</td>
<td align="right">0.9620527</td>
<td align="right">0.9637110</td>
<td align="right">0.9513490</td>
<td align="right">0.9568884</td>
<td align="right">1.0000000</td>
<td align="right">0.9658420</td>
<td align="right">0.9707951</td>
<td align="right">0.9582526</td>
<td align="right">0.9607596</td>
</tr>
<tr class="even">
<td>Combined.JG</td>
<td align="right">0.9814294</td>
<td align="right">0.9636047</td>
<td align="right">0.9611764</td>
<td align="right">0.9465928</td>
<td align="right">0.9744784</td>
<td align="right">0.9602197</td>
<td align="right">0.9589687</td>
<td align="right">0.9694140</td>
<td align="right">0.9504304</td>
<td align="right">0.9838795</td>
<td align="right">0.9658420</td>
<td align="right">1.0000000</td>
<td align="right">0.9740497</td>
<td align="right">0.9872414</td>
<td align="right">0.9680140</td>
</tr>
<tr class="odd">
<td>Combined.DHX</td>
<td align="right">0.9586613</td>
<td align="right">0.9614223</td>
<td align="right">0.9555638</td>
<td align="right">0.9560525</td>
<td align="right">0.9748043</td>
<td align="right">0.9883522</td>
<td align="right">0.9871373</td>
<td align="right">0.9697808</td>
<td align="right">0.9575416</td>
<td align="right">0.9650818</td>
<td align="right">0.9707951</td>
<td align="right">0.9740497</td>
<td align="right">1.0000000</td>
<td align="right">0.9657162</td>
<td align="right">0.9902138</td>
</tr>
<tr class="even">
<td>Healthy.JG_1</td>
<td align="right">0.9610586</td>
<td align="right">0.9548696</td>
<td align="right">0.9545264</td>
<td align="right">0.9414062</td>
<td align="right">0.9652938</td>
<td align="right">0.9537794</td>
<td align="right">0.9528822</td>
<td align="right">0.9624151</td>
<td align="right">0.9450088</td>
<td align="right">0.9630776</td>
<td align="right">0.9582526</td>
<td align="right">0.9872414</td>
<td align="right">0.9657162</td>
<td align="right">1.0000000</td>
<td align="right">0.9589588</td>
</tr>
<tr class="odd">
<td>Affected_3</td>
<td align="right">0.9534496</td>
<td align="right">0.9560137</td>
<td align="right">0.9495342</td>
<td align="right">0.9413015</td>
<td align="right">0.9672163</td>
<td align="right">0.9675438</td>
<td align="right">0.9664420</td>
<td align="right">0.9597859</td>
<td align="right">0.9443274</td>
<td align="right">0.9590722</td>
<td align="right">0.9607596</td>
<td align="right">0.9680140</td>
<td align="right">0.9902138</td>
<td align="right">0.9589588</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code># Plot
heatmap.2(CorMat, trace=&quot;none&quot;)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, as expected. Biological replicates from same line cluster together. From here on forth I will deal with just the combined samples.</p>
<pre class="r"><code>#Subset the combined samples
CountTable.matrix &lt;- CountTable.matrix[, grep(&quot;Combined*&quot;, colnames(CountTable.matrix))]

dim(CountTable.matrix)</code></pre>
<pre><code>[1] 58676     4</code></pre>
<p>And do the same filtering as before, filtering out genes below a threshold for cpm expression…</p>
<pre class="r"><code># Pick a cutoff for lowly expressed genes to exclude
hist(median_log2_cpm)
expr_cutoff &lt;- 2
abline(v = expr_cutoff, col = &quot;red&quot;, lwd = 3)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#How many genes pass this threshold
sum(median_log2_cpm &gt; expr_cutoff)</code></pre>
<pre><code>[1] 11978</code></pre>
<pre class="r"><code>CountTable.matrix.filtered &lt;- CountTable.matrix[median_log2_cpm &gt; expr_cutoff, ]

# Recalculate cpm from filtered count table
cpm &lt;- cpm(CountTable.matrix.filtered, log=TRUE, prior.count=0.5)

# look at correlations from filtered cpm matrix
cor(cpm) %&gt;%
heatmap.2(., trace=&quot;none&quot;, cexRow=1, cexCol=1)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Also look at similarity with MDS plot
plotMDS(cpm)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Interestingly, samples do not primarily segregate by condition (healthy vs affected). In other words, biological noise (effects not due to the healthy vs affected contrast of interest) are greater than the healthy vs affected. I wil still do the differential expression analysis as planned, since it will yield a list of DE genes that are more likely truly due to the DHX38 mutations, but nonetheless it is likely that a large fraction of the DE genes we find will be due to effects not due to the DHX38 genotype but rather other line-specific effects, since these other line-specific effects have a generally stronger effect. In other words, the hypothesized effect of DHX38 mutations are quite small.</p>
<p>Now start prepping data for differential expression analysis</p>
<pre class="r"><code>colnames(CountTable.matrix.filtered)</code></pre>
<pre><code>[1] &quot;Combined.BIH4&quot;  &quot;Combined.XM001&quot; &quot;Combined.JG&quot;    &quot;Combined.DHX&quot;  </code></pre>
<pre class="r"><code># A for affected, H for healthy
group &lt;- c(&quot;H&quot;, &quot;H&quot;, &quot;H&quot;, &quot;A&quot;)


# y is a DGELIst object contains count table and other metadata
y &lt;- DGEList(counts = CountTable.matrix.filtered, group = group)

# calculate normalization factors using default TMM method which more accurately scales samples with different library sizes, for example while considering that highly expressed genes between conditions can have undesired effects if just using simple CountsPerMillion normalization.

y &lt;- calcNormFactors(y)
y$samples %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">group</th>
<th align="right">lib.size</th>
<th align="right">norm.factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Combined.BIH4</td>
<td align="left">H</td>
<td align="right">40494235</td>
<td align="right">1.0134834</td>
</tr>
<tr class="even">
<td>Combined.XM001</td>
<td align="left">H</td>
<td align="right">60272758</td>
<td align="right">0.9668125</td>
</tr>
<tr class="odd">
<td>Combined.JG</td>
<td align="left">H</td>
<td align="right">55267740</td>
<td align="right">1.0171157</td>
</tr>
<tr class="even">
<td>Combined.DHX</td>
<td align="left">A</td>
<td align="right">89524716</td>
<td align="right">1.0033922</td>
</tr>
</tbody>
</table>
<p>Now estimate common, trended, and tagwise dispersion, and test for differential expression</p>
<pre class="r"><code>#estimate dispersion, common, trended, and tag-wise (a combination of gene-specific dispersion and trend)
y &lt;- estimateDisp(y)</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>sqrt(y$common.dispersion) # biological coefficient of variation</code></pre>
<pre><code>[1] 0.2478897</code></pre>
<pre class="r"><code>#Plot dispersions
plotBCV(y)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># by default, the tagwise dispersion is used for DE testing.
# Perform testing
et &lt;- exactTest(y, pair=c(&quot;H&quot;,&quot;A&quot;))
results_edgeR &lt;- topTags(et, n = nrow(CountTable.matrix.filtered), sort.by = &quot;none&quot;)

head(results_edgeR$table) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000279457</td>
<td align="right">0.0008339</td>
<td align="right">2.323825</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>ENSG00000248527</td>
<td align="right">-0.6449759</td>
<td align="right">2.974398</td>
<td align="right">0.5288493</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>ENSG00000228794</td>
<td align="right">-0.1405206</td>
<td align="right">3.373410</td>
<td align="right">0.8252138</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>ENSG00000230699</td>
<td align="right">0.3685152</td>
<td align="right">2.468252</td>
<td align="right">0.5953829</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>ENSG00000188976</td>
<td align="right">-0.0582193</td>
<td align="right">6.968201</td>
<td align="right">0.8141212</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>ENSG00000187961</td>
<td align="right">-0.2584633</td>
<td align="right">4.302562</td>
<td align="right">0.5637095</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#histogram of nominal P-vals
hist(results_edgeR$table$PValue)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># how many genes below FDR threshold
sum(results_edgeR$table$FDR &lt; .1)</code></pre>
<pre><code>[1] 187</code></pre>
<pre class="r"><code>#MA plot showing FDR&lt;0.1
plotSmear(et, de.tags = rownames(results_edgeR)[results_edgeR$table$FDR &lt; .01])
abline(h = c(-2, 2), col = &quot;blue&quot;)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok we did differential testing. But are the significant P-values still just due to technical factors that would be present in any 1v3 contrast of genetically independent cell lines? Let’s try the permutations of the sample labels to see if there is inflation of small P-values in the real contrast compared to the permutated contrasts (For example, the 1v3 comparison of [BIH4] vs [DHX39, XM001, JG])?</p>
<pre class="r"><code>DiffExpTest &lt;- function(groups){
  Results &lt;- DGEList(counts = CountTable.matrix.filtered, group = groups) %&gt;%
  estimateDisp() %&gt;%
  exactTest(pair=c(&quot;H&quot;, &quot;A&quot;))
  return(Results$table$PValue)
}

RealResults &lt;- DiffExpTest(groups=c(&quot;H&quot;, &quot;H&quot;, &quot;H&quot;, &quot;A&quot;))</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>Permutation1Results &lt;- DiffExpTest(groups=c(&quot;A&quot;, &quot;H&quot;, &quot;H&quot;, &quot;H&quot;))</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>Permutation2Results &lt;- DiffExpTest(groups=c(&quot;H&quot;, &quot;A&quot;, &quot;H&quot;, &quot;H&quot;))</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>Permutation3Results &lt;- DiffExpTest(groups=c(&quot;H&quot;, &quot;H&quot;, &quot;A&quot;, &quot;H&quot;))</code></pre>
<pre><code>Design matrix not provided. Switch to the classic mode.</code></pre>
<pre class="r"><code>data.frame(RealResults, Permutation1Results, Permutation2Results, Permutation3Results) %&gt;% gather(value=&quot;Actual.P&quot;) %&gt;%
  group_by(key) %&gt;%
  mutate(Expected.P=percent_rank(Actual.P)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(color=key, x=-log10(Expected.P), y=-log10(Actual.P))) +
  geom_point() +
  geom_abline() +
  scale_color_manual(labels = c(Permutation1Results=&quot;Permuted; [BIH4] v [DHX, XM001, JG]&quot;,
                                Permutation2Results=&quot;Permuted; [XM001] v [DHX, BIH4, JG]&quot;,
                                Permutation3Results=&quot;Permuted; [JG] v [DHX, XM001, BIH4]&quot;,
                                RealResults=&quot;Actual; [DHX] v [BIH4, XM001, JG]&quot;),
                     values=c(&quot;red&quot;, &quot;blue&quot;, &quot;purple&quot;, &quot;black&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, as I suggested before, the possible biological effect of DHX38 mutations is relatively small, compared to inter-line variation between non-DHX38 mutant samples. For example, the [XM001] v [DHX, BIH4, JG] comparison has more inflated P-values. So most of the significant changes we see are still likely due to non-DHX38 related things.</p>
<p>In any case, let’s still explore the results from the original contrast.</p>
<p>First add common gene names to the results table</p>
<pre class="r"><code>#Use biomart to get common gene names and ensembl IDs
human_mart &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)

MyBioMartQuery &lt;- getBM(attributes = c(&quot;ensembl_gene_id&quot;,&quot;external_gene_name&quot;, &quot;description&quot;), mart=human_mart, filters=&quot;ensembl_gene_id&quot;, values=rownames(results_edgeR))

results.more.descriptive &lt;- results_edgeR$table %&gt;%
  rownames_to_column(&quot;ensembl_gene_id&quot;) %&gt;%
  left_join(MyBioMartQuery, by=&quot;ensembl_gene_id&quot;)

head(results.more.descriptive) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
<th align="left">external_gene_name</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000279457</td>
<td align="right">0.0008339</td>
<td align="right">2.323825</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="left">WASH9P</td>
<td align="left">WAS protein family homolog 9, pseudogene [Source:HGNC Symbol;Acc:HGNC:53981]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000248527</td>
<td align="right">-0.6449759</td>
<td align="right">2.974398</td>
<td align="right">0.5288493</td>
<td align="right">1</td>
<td align="left">MTATP6P1</td>
<td align="left">MT-ATP6 pseudogene 1 [Source:HGNC Symbol;Acc:HGNC:44575]</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000228794</td>
<td align="right">-0.1405206</td>
<td align="right">3.373410</td>
<td align="right">0.8252138</td>
<td align="right">1</td>
<td align="left">LINC01128</td>
<td align="left">long intergenic non-protein coding RNA 1128 [Source:HGNC Symbol;Acc:HGNC:49377]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000230699</td>
<td align="right">0.3685152</td>
<td align="right">2.468252</td>
<td align="right">0.5953829</td>
<td align="right">1</td>
<td align="left">AL645608.2</td>
<td align="left">novel transcript</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000188976</td>
<td align="right">-0.0582193</td>
<td align="right">6.968201</td>
<td align="right">0.8141212</td>
<td align="right">1</td>
<td align="left">NOC2L</td>
<td align="left">NOC2 like nucleolar associated transcriptional repressor [Source:HGNC Symbol;Acc:HGNC:24517]</td>
</tr>
<tr class="even">
<td align="left">ENSG00000187961</td>
<td align="right">-0.2584633</td>
<td align="right">4.302562</td>
<td align="right">0.5637095</td>
<td align="right">1</td>
<td align="left">KLHL17</td>
<td align="left">kelch like family member 17 [Source:HGNC Symbol;Acc:HGNC:24023]</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Check if DHX38 is differentially expressed
DHX38.results &lt;- results.more.descriptive %&gt;%
  filter(external_gene_name==&quot;DHX38&quot;)

kable(DHX38.results)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
<th align="left">external_gene_name</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000140829</td>
<td align="right">0.0673342</td>
<td align="right">6.889669</td>
<td align="right">0.7861621</td>
<td align="right">1</td>
<td align="left">DHX38</td>
<td align="left">DEAH-box helicase 38 [Source:HGNC Symbol;Acc:HGNC:17211]</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Plot smear with DHX label
PlotSmear &lt;- results.more.descriptive %&gt;%
  mutate(
    Significance=case_when(
      FDR&lt;0.1 ~ &quot;FDR&lt;0.1&quot;,
      FDR&gt;0.1 ~ &quot;Not significant&quot;,
      external_gene_name==&quot;DHX38&quot; ~ &quot;DHX38&quot;),
    Label=case_when(
      external_gene_name==&quot;DHX38&quot; ~ paste0(&quot;DHX38\nP=&quot;, signif(DHX38.results$PValue, 3)),
      !external_gene_name==&quot;DHX38&quot; ~ &quot;&quot;),
  ) %&gt;% 
  arrange(desc(FDR))

DHX38.results &lt;- PlotSmear %&gt;%
  filter(external_gene_name==&quot;DHX38&quot;)

ggplot(PlotSmear, aes(x=logCPM, y=logFC, color=Significance, label=Label)) +
    geom_point() +
    geom_point(data=DHX38.results, color=&quot;black&quot;, show.legend = F) +
    geom_text_repel(data=DHX38.results, color=&quot;black&quot;, show.legend = F) +
    geom_hline(yintercept = 0, linetype=&quot;dashed&quot;) +
    theme_bw() +
    theme(legend.title = element_blank())</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok good. DHX38 is not differentially expressed. So if the alleles have some affect on splicing through DHX38, is not likely through modulating expression of DHX38 levels, but rather altering function.</p>
<p>Let’s make some plots to get to know significance testing results a little bit more… Keep in mind we don’t know much about the types of replicates these are (technical, versus biological split before differentiation, split after differentiation, etc), and that all these are from a single healthy and a single affected patient.</p>
<p>Volcano plot:</p>
<pre class="r"><code>results.more.descriptive %&gt;%
  mutate(Significance=FDR&lt;0.1) %&gt;%
  ggplot(aes(x=logFC, y=-log10(PValue), color=Significance)) +
  geom_point(alpha=0.2) +
  theme_bw()</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot of original results of a significant gene. Useful to check polarization of the logFC values (I think I set up the <code>exactTest</code> function to be such that positive logFC means more highly expressed in affected, but let’s double check that)</p>
<pre class="r"><code>Top5DE_Genes &lt;- results.more.descriptive %&gt;%
  arrange(FDR) %&gt;% head(5) %&gt;%
  pull(ensembl_gene_id)

results.more.descriptive %&gt;%
  filter(ensembl_gene_id %in% Top5DE_Genes) %&gt;%
  dplyr::select(ensembl_gene_id, external_gene_name, logFC) %&gt;%
  kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ensembl_gene_id</th>
<th align="left">external_gene_name</th>
<th align="right">logFC</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000123562</td>
<td align="left">MORF4L2</td>
<td align="right">1.289187</td>
</tr>
<tr class="even">
<td align="left">ENSG00000148841</td>
<td align="left">ITPRIP</td>
<td align="right">2.426031</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237550</td>
<td align="left">AC243919.1</td>
<td align="right">-4.985699</td>
</tr>
<tr class="even">
<td align="left">ENSG00000215030</td>
<td align="left">RPL13P12</td>
<td align="right">3.646536</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000132872</td>
<td align="left">SYT4</td>
<td align="right">1.850729</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cpm[Top5DE_Genes,] %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;ensembl_gene_id&quot;) %&gt;%
  gather(key=&quot;sample&quot;, value=&quot;log2cpm&quot;, -ensembl_gene_id) %&gt;% 
  ggplot(aes(x=sample, y=log2cpm)) +
  geom_col() +
  facet_wrap(~ensembl_gene_id) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok correct, positive FC means higher expressed in affected samples. Now let’s do gene set enrichment analysis (GSEA) to see which types of gene categories are enriched for differentially expressed genes. For this we need an ordered list of genes. We could use absolute value of log-fold change, or maybe -log10(Pvalue). Judging from the volcano plot I’d guess the results would be virtually identical either way</p>
<pre class="r"><code>#had to add very small number to Pvalue because very significant genes have Pvalues that are 0
gsea.results &lt;- results.more.descriptive %&gt;%
  mutate(RankingMetric = -log10(PValue+1E-300)) %&gt;%
  arrange(desc(RankingMetric)) %&gt;%
  dplyr::select(ensembl_gene_id, RankingMetric) %&gt;%
  deframe() %&gt;%
  gseGO(ont=&quot;ALL&quot;, OrgDb=org.Hs.eg.db, keyType=&#39;ENSEMBL&#39;, nPerm=500000)</code></pre>
<p>Now let’s explore the GSEA results</p>
<pre class="r"><code>gsea.results.df &lt;- as.data.frame(gsea.results)
gsea.results.df %&gt;% 
  group_by(ONTOLOGY) %&gt;%
  filter(rank(p.adjust, ties.method=&quot;first&quot;)&lt;=8) %&gt;%
  ggplot(aes(x=enrichmentScore, y=Description, color=p.adjust, size=setSize)) +
  geom_point() +
  facet_grid(ONTOLOGY~., scales = &quot;free&quot;) +
  scale_colour_viridis_c(trans=&quot;log10&quot;, limits=c(2E-4, 1), direction=-1, option=&quot;D&quot;) +
  xlab(&quot;Enrichment&quot;) +
  scale_y_discrete(labels = function(x) lapply(strwrap(x, width = 60, simplify = FALSE), paste, collapse=&quot;\n&quot;)) +
  labs(color = &quot;Adjusted P-value&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gsea.results.df %&gt;%
  arrange(p.adjust) %&gt;% head(10) %&gt;%
  dplyr::select(ID, Description, ONTOLOGY, p.adjust, setSize, enrichmentScore)</code></pre>
<pre><code>           ID                                    Description ONTOLOGY
1  GO:0000904 cell morphogenesis involved in differentiation       BP
2  GO:0003002                                regionalization       BP
3  GO:0005887          integral component of plasma membrane       CC
4  GO:0006935                                     chemotaxis       BP
5  GO:0007389                  pattern specification process       BP
6  GO:0007409                                   axonogenesis       BP
7  GO:0007411                                  axon guidance       BP
8  GO:0009952       anterior/posterior pattern specification       BP
9  GO:0010720        positive regulation of cell development       BP
10 GO:0021953  central nervous system neuron differentiation       BP
       p.adjust setSize enrichmentScore
1  0.0003977492     473       0.5906860
2  0.0003977492     190       0.6797062
3  0.0003977492     462       0.5818294
4  0.0003977492     281       0.6042926
5  0.0003977492     243       0.6538520
6  0.0003977492     294       0.6323043
7  0.0003977492     149       0.6738251
8  0.0003977492     104       0.7179275
9  0.0003977492     355       0.5840187
10 0.0003977492     117       0.7023207</code></pre>
<pre class="r"><code>gsea.results.df %&gt;%
  arrange(desc(enrichmentScore)) %&gt;% head(10) %&gt;%
  dplyr::select(ID, Description, ONTOLOGY, p.adjust, setSize)</code></pre>
<pre><code>           ID                                            Description ONTOLOGY
1  GO:0048665                              neuron fate specification       BP
2  GO:0060579        ventral spinal cord interneuron fate commitment       BP
3  GO:0060581 cell fate commitment involved in pattern specification       BP
4  GO:0021514        ventral spinal cord interneuron differentiation       BP
5  GO:0071625                                  vocalization behavior       BP
6  GO:0032332     positive regulation of chondrocyte differentiation       BP
7  GO:0060512                           prostate gland morphogenesis       BP
8  GO:0060740                prostate gland epithelium morphogenesis       BP
9  GO:0048663                                 neuron fate commitment       BP
10 GO:0021513                  spinal cord dorsal/ventral patterning       BP
      p.adjust setSize
1  0.012532160      14
2  0.036769704      10
3  0.036769704      10
4  0.032182086      11
5  0.040837996      13
6  0.044858667      13
7  0.032515162      15
8  0.032515162      15
9  0.001242509      29
10 0.040837996      15</code></pre>
<p>Lot’s of interesting stuff. Let’s write out the full table of results. Here is a plot of a few terms that catch’s the eye (<a href="GO:0007409" class="uri">GO:0007409</a>, axonogenesis). Though, another term that might point towards the interpretation of these differences being more related to the ipsc differentiation differences with n=1 (or maybe n=3; I still don’t fully understand the experiment design).</p>
<p>This plot captures the essence of how GSEA works. A rug depicting instances of genes belonging to sets is shown on x-axis. Some gene sets are enriched for genes at the top of the ordered list, leading to more enrichment than you expect by chance as you move from the top (left) to bottom (right) of the ordered list.</p>
<pre class="r"><code>gseaplot2(gsea.results, geneSetID = c(&quot;GO:0000904&quot;, &quot;GO:0007409&quot;, &quot;GO:0060579&quot;), pvalue_table = F)</code></pre>
<p><img src="figure/20200110_DiffExpression_MoreReplicates.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Another thing to keep in mind with this and any GSEA is that there is a lot of overlap in some gene set terms, as is clear above in the rug plot along the x-axis. Some sets are huge (hundreds of terms).</p>
<pre class="r"><code>#Write out gsea results
write.table(gsea.results.df, file=&quot;../output/GeneExpression.gsea.WithMoreReplicates.tab&quot;, quote=F, sep=&#39;\t&#39;, row.names = F)

#write differential expression results
write.table(results.more.descriptive, file=&quot;../output/GeneExpression.DE.WithMoreReplicates.tab&quot;, quote=F, sep=&#39;\t&#39;, row.names = F)</code></pre>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<p>Perhaps some neuronal related gene sets enriched, though with small sample sizes and confounding DHX38 genotype with other individual genetic differences in this study design, it is difficult to draw strong conclusions about the effect of these DHX38 mutations.</p>
<p>At the very least we can be sure DHX38 is not very differently expresesed.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] enrichplot_1.4.0       org.Hs.eg.db_3.8.2     AnnotationDbi_1.46.1  
 [4] IRanges_2.18.3         S4Vectors_0.22.1       Biobase_2.44.0        
 [7] BiocGenerics_0.30.0    clusterProfiler_3.12.0 ggrepel_0.8.1         
[10] biomaRt_2.40.5         gplots_3.0.1.1         corrplot_0.84         
[13] edgeR_3.26.8           limma_3.40.6           knitr_1.26            
[16] forcats_0.4.0          stringr_1.4.0          dplyr_0.8.3           
[19] purrr_0.3.3            readr_1.3.1            tidyr_1.0.0           
[22] tibble_2.1.3           ggplot2_3.2.1          tidyverse_1.3.0       

loaded via a namespace (and not attached):
  [1] fgsea_1.10.1        colorspace_1.4-1    ellipsis_0.3.0     
  [4] ggridges_0.5.1      rprojroot_1.3-2     qvalue_2.16.0      
  [7] fs_1.3.1            rstudioapi_0.10     farver_2.0.1       
 [10] urltools_1.7.3      graphlayouts_0.5.0  bit64_0.9-7        
 [13] fansi_0.4.0         lubridate_1.7.4     xml2_1.2.2         
 [16] splines_3.6.1       GOSemSim_2.10.0     polyclip_1.10-0    
 [19] zeallot_0.1.0       jsonlite_1.6        workflowr_1.5.0    
 [22] broom_0.5.2         GO.db_3.8.2         dbplyr_1.4.2       
 [25] ggforce_0.3.1       BiocManager_1.30.10 compiler_3.6.1     
 [28] httr_1.4.1          rvcheck_0.1.7       backports_1.1.5    
 [31] assertthat_0.2.1    Matrix_1.2-18       lazyeval_0.2.2     
 [34] cli_2.0.0           later_1.0.0         tweenr_1.0.1       
 [37] htmltools_0.4.0     prettyunits_1.0.2   tools_3.6.1        
 [40] igraph_1.2.4.2      gtable_0.3.0        glue_1.3.1         
 [43] reshape2_1.4.3      DO.db_2.9           fastmatch_1.1-0    
 [46] Rcpp_1.0.2          cellranger_1.1.0    vctrs_0.2.0        
 [49] gdata_2.18.0        nlme_3.1-143        ggraph_2.0.0       
 [52] xfun_0.11           rvest_0.3.5         lifecycle_0.1.0    
 [55] gtools_3.8.1        XML_3.98-1.20       DOSE_3.10.2        
 [58] europepmc_0.3       MASS_7.3-51.4       scales_1.1.0       
 [61] tidygraph_1.1.2     hms_0.5.2           promises_1.1.0     
 [64] RColorBrewer_1.1-2  curl_4.3            yaml_2.2.0         
 [67] memoise_1.1.0       gridExtra_2.3       UpSetR_1.4.0       
 [70] triebeard_0.3.0     stringi_1.4.3       RSQLite_2.1.4      
 [73] highr_0.8           caTools_1.17.1.3    BiocParallel_1.18.1
 [76] rlang_0.4.1         pkgconfig_2.0.3     bitops_1.0-6       
 [79] evaluate_0.14       lattice_0.20-38     labeling_0.3       
 [82] cowplot_1.0.0       bit_1.1-14          tidyselect_0.2.5   
 [85] plyr_1.8.5          magrittr_1.5        R6_2.4.1           
 [88] generics_0.0.2      DBI_1.0.0           pillar_1.4.2       
 [91] haven_2.2.0         withr_2.1.2         RCurl_1.95-4.12    
 [94] modelr_0.1.5        crayon_1.3.4        KernSmooth_2.23-16 
 [97] rmarkdown_1.18      viridis_0.5.1       progress_1.2.2     
[100] locfit_1.5-9.1      grid_3.6.1          readxl_1.3.1       
[103] data.table_1.12.8   blob_1.2.0          git2r_0.26.1       
[106] reprex_0.3.0        digest_0.6.23       httpuv_1.5.2       
[109] gridGraphics_0.4-1  munsell_0.5.0       ggplotify_0.0.4    
[112] viridisLite_0.3.0  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
