---
title: "Untitled"
author: "Benjamin Fair"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
  html_document:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)
```

## Introduction

Exploratory differential expression analysis comparing RNA-seq from iPSC derived neurons from healthy control vs patient with biallelic mutations in DHX38 that may affect splicing and gene expression. Some things I want to check is what kinds of gene sets are differentially spliced, as well as if there are certain intron features associated with differentially spliced introns (weaker 5'ss motif perhaps)

## Analysis

Raw RNA-seq data has been aligned and collapsed into leafcutter splice junction count table (rows of introns, columns of samples, each cell is exonic read counts) using code in `code` section of this repo. Also, leafcutter has been run to identify differentially spliced clusters of splice junctions, creating output files in `output`. Here I will explore that output and maybe some other exploratory analyses.

First, load necessary libraries
```{r}
library(tidyverse)
library(knitr)
library(edgeR)
library(biomaRt)
library("clusterProfiler")
library("org.Hs.eg.db")
library(enrichplot)
library(psych)
library(corrplot)
```

load in leafcutter output

```{r}
sig <- read.table("../output/differential_splicing_significance.txt.gz", sep='\t', header=T)

head(sig) %>% kable()

effect_sizes <- read.table("../output/differential_splicing_effect_sizes.txt.gz", sep='\t', header=T)

head(effect_sizes) %>% kable()
```

Get a sense of number of intron clusters tested, number significant, etc

```{r}
#how many clusters tested ("Success")
table(sig$status)

#histogram of Pvalues
hist(sig$p)

#how many significant (p.adjust < 0.1)
table(sig$p.adjust<0.1)

# merge significance and cluster tables
leafcutter.merged <- effect_sizes %>%
  mutate(cluster=gsub("(.+?:).+?:.+?:(clu.+?)", "\\1\\2", intron, perl=T)) %>%
  mutate(junc_id=gsub("(.+?:.+?:.+?):clu.+", "\\1", intron, perl=T)) %>%
  left_join(sig, by="cluster")

# volcano plot of cluster-pvalues and largest within-cluster-delta-psi
leafcutter.merged %>%
  mutate(abs.deltapsi = abs(deltapsi)) %>%
  group_by(cluster) %>% 
  top_n(n=1, abs.deltapsi) %>%
  mutate(sig=p.adjust<0.05) %>%
  ggplot(aes(x=deltapsi, y=-log10(p), color=sig)) +
    geom_point(alpha=0.05) +
    theme_bw()
```

let's do a gsea to see what genes have differentially spliced clusters of introns. For this, we need an ordered gene list, with each gene appearing once. Since the biological question related to this is what are the genes or gene sets that are getting disrupted to cause the organismal phenotype, i think it is important to consider not just significance (P-value) but also effect size, perhaps as measured by delta-psi. Since significane and effect size can be quite different in a splicing analysis (see volcano plot above) because of low read counts on some junctions, let's first filter out all non-significant splicing changes (to filter out the noisy effect size estimates) Then let's make an ordered list based on the maximum delta-PSI for each gene.

```{r}
#let's use a fairly permissive significance threshold to be more inclusive. how about FDR>0.3
hist(leafcutter.merged$p.adjust)

#filter and get max delta-psi for each gene. Keep in mind we need to filter out non-valid genes for the clusters or introns that have more than one gene listed in the genes column
OrderedGeneList <- leafcutter.merged %>%
  filter(p.adjust<0.3) %>%
  mutate(abs.deltapsi = abs(deltapsi)) %>%
  group_by(genes) %>% 
  top_n(n=1, abs.deltapsi) %>%
  filter(!grepl(",",genes)) %>%
  arrange(desc(abs.deltapsi)) %>%
  dplyr::select(genes, abs.deltapsi) %>%
  deframe()

head(OrderedGeneList)

gsea.results <- gseGO(OrderedGeneList, ont="ALL", OrgDb=org.Hs.eg.db, keyType='SYMBOL', nPerm=100000)

gsea.results.df <- as.data.frame(gsea.results)

dim(gsea.results.df)

```

No significant GSEA results at the default P.adjust<0.5 threshold.

Ok, besides identifying the genes (or kinds of genes) through which the disease phenotype is potentially mediated, maybe the other questions I should be asking is what kinds of intron features distinguish differentially spliced introns, which may be more enlightening about general splicing mechanisms in these DHX38 mutants.

I have scored the 5'ss and 3'ss motifs for each intron tested using a position specific score matrix based on annotated introns. I additionally used [svm-bpfinder](http://regulatorygenomics.upf.edu/Software/SVM_BP/) to score the best computationally predicted branch for each intron. I also annotated whether each intron is annotated, alt3'ss, alt5'ss, new pairing of splice sites, or new intron. All of that work is hidden in the `code` section and output into a file in `output`. 

Let's see if any of those features correlate with differentially expressed introns. If there is the DHX38 mutations preferentially activates less consensus splice sites for example, I expect the directionality to be consistent. Therefore, I will correlate those scores with either the logFC for each intron, for significant introns.

```{r}
#read intron features
IntronFeatures <- read.table("../output/IntronFeatures.txt.gz", sep='\t', header=T)

head(IntronFeatures) %>% kable()

# Check that the number of types of unannotated splicing is similar in both strands. Sometimes bugs related to coordinates produce wonky results
IntronFeatures %>%
  filter(strand=="-") %>%
  pull(type) %>% table()

IntronFeatures %>%
  filter(strand=="+") %>%
  pull(type) %>% table()

# merge the intron features with the leafcutter results and filter for the significantly differentially spliced introns that have the greatest effect size within cluster
leafcutter.merged.w.features <- leafcutter.merged %>%
  separate(junc_id, c("chrom", "start", "stop"), sep = ":") %>%
  mutate(newChrom=sub("chr", "", chrom),
         newStop=as.numeric(stop)-1,
         strand=str_sub(intron, start=-1)) %>%
  mutate(IntronID=str_c(newChrom, start, newStop, strand, sep="_")) %>%
  dplyr::select(-c("chrom", "start", "stop", "newStop", "newChrom")) %>%
  mutate(abs.effectsize = abs(logef)) %>%
  group_by(cluster) %>% 
  top_n(n=1, abs.effectsize) %>%
  ungroup() %>%
  filter(p.adjust<0.2) %>%
  left_join(IntronFeatures, by=c("IntronID"="Intron"))

# Are specific types of unannotated splicing more prevalent?
leafcutter.merged.w.features %>%
  pull(type) %>% table()

colnames(leafcutter.merged.w.features)

# Set up features and effect sizes into a cleaner data frame
IntronFeaturesAndEffectSizes <- leafcutter.merged.w.features %>%
  dplyr::select(IntronID, logef, DonorScore, AcceptorScore, ss_dist, bp_scr, ppt_len, ppt_scr, svm_scr) %>%
  column_to_rownames("IntronID")

# See if there is any correlations
CorrelationsWithFeatures <- corr.test(x=IntronFeaturesAndEffectSizes[,1],
                         y=IntronFeaturesAndEffectSizes[,-1],
                         method="spearman",
                         adjust="none")

# table of spearmans r, pvalue, and n
data.frame(r= t(CorrelationsWithFeatures$r), p=t(CorrelationsWithFeatures$p), n=t(CorrelationsWithFeatures$n)) %>% kable()
```

None of the features tested seem to correlate with the effect size of largest-effect introns in significant clusters.


Check if the significant positive effect-size splicing changes have more cryptic 5'ss, or more cryptic 3'ss than negative effect-size splicing changes.

When doing this analysis, some people think in terms of deltapsi, some people might think in terms of a different metric of effect size. Let's look at both.


```{r}
# leafcutter's logef effect size, versus deltapsi
qplot(leafcutter.merged.w.features$logef, leafcutter.merged.w.features$deltapsi) + theme_bw()

#significant positive deltapsi introns
leafcutter.merged.w.features %>%
  filter(deltapsi > 0) %>%
  pull(type) %>% table()

#significant negative deltapsi introns
leafcutter.merged.w.features %>%
  filter(deltapsi < 0) %>%
  pull(type) %>% table()

leafcutter.merged.w.features %>%
  ggplot(aes(x=deltapsi, color=type)) +
    stat_ecdf(geom = "step") +
    ylab("Cumulative frequency") +
    theme_bw()
```

Maybe 5'ss stick out? Let's test it more explicitly.


The null hypothesis here is that any given type of splicing will not significantly deviate from median of 0 effect size (similar number of positive effects as negative effects). Wilcox.test to test with null as mu=0 should work.


```{r}
wilcox.test(
  leafcutter.merged.w.features %>%
  filter(type=="Alt5ss") %>% pull(deltapsi)
    )

wilcox.test(
  leafcutter.merged.w.features %>%
  filter(type=="Alt5ss") %>% pull(logef)
    )

```

Hmm. maybe there is something here. Considering all the exploration I've been doing, and without multiple hypothesis correction in many cases this might or might not be real.

## TODO
- should look at the alt 5'ss and see if there is any distance constraint with the annotated 5'ss, similar to the SF3B1 mutants that have a very specific distance constraint to the usual splice sites that get activated

## Conclusions


