---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

Welcome to my research website.

## Initial analysis

3 replicates (same cell line at different passages) of iPSC derived neural progenitor cells from patient with DHX38 biallelic mutations, compared with 3 replicates of an unrelated healthy control.

- [Differential gene expression](20191203_DiffExpression.html)
- [Plot differential expression of splicing genes](20191216_DifferentialExpression_Splicing.html)
- [Differential splicing](20191203_DifferentialSplicing.html)
- [Differential splicing pos control analysis](20191206_DifferentialSplicingControl.html)
- [Check overlap with polyA length](20191216_polyA_Intersect.html)

## Analysis with more replicates

Here I did some of the same analyses, simply copy/pasting a lot of code used to in the analyses above. However, now we have a slightly different experimental design. We still have the 3 DHX38 iPSC-derived NPC replicates (same differentiation, replicated at different passages), and the 3 healthy control replicates (cell line XM001). But now we additionally have 2 replicates of cell line BIH4 and 2 replicates of cell line JG (both of these are also healthy controls, but with different genetic background and different differentiation, where the replicates still represent collections on different passage days). Because we have different levels of replication (we have multiple healthy lines, each of which has replicates), this poses a new challange to the differential analyses, since not all samples are independent meseasurements. One way to solve this problem is to use mixed models. But leafcutter wasn't built to do this easily. Another reasonable way is to simply combine the passage replicates and then simply do a 1 affected vs 3 healthy contrast.

In the first analyses, I repeat the differential expression analysis I did previously. Part of this involves showing that the replicates of the same cell line at different passages are highly correlated compared to different cell lines, which somewhat justifies combining these. In the next analyses, I repeat the splicing analysis above but now using this 1vs3 design.

- [Repeat of differential expression analysis, but with 1v3](20200110_DiffExpression_MoreReplicates.html)
- [Repeat of splicing analyses, but with 1v3 design](20190105_DifferentialSplicingMoreReplicates.html)

